{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c2df53bd-00ad-47d0-bd8c-dda94e076a87",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:18.428427Z",
     "iopub.status.busy": "2022-09-26T01:28:18.427731Z",
     "iopub.status.idle": "2022-09-26T01:28:18.445958Z",
     "shell.execute_reply": "2022-09-26T01:28:18.444865Z",
     "shell.execute_reply.started": "2022-09-26T01:28:18.428281Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "\n",
    "sys.path.append(\"../../vae_zinb_reprn/\")\n",
    "sys.path.append(\"../src/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "61d05ab2-879b-47d6-9040-9832b3fa7d7a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:19.132057Z",
     "iopub.status.busy": "2022-09-26T01:28:19.131437Z",
     "iopub.status.idle": "2022-09-26T01:28:19.791969Z",
     "shell.execute_reply": "2022-09-26T01:28:19.790829Z",
     "shell.execute_reply.started": "2022-09-26T01:28:19.132005Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import datetime\n",
    "import logging\n",
    "import os\n",
    "import time\n",
    "from functools import cached_property\n",
    "\n",
    "from sklearn.metrics import average_precision_score, ndcg_score, roc_auc_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7b100ceb-433d-4749-8076-20b085d23714",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:20.113763Z",
     "iopub.status.busy": "2022-09-26T01:28:20.112945Z",
     "iopub.status.idle": "2022-09-26T01:28:20.168842Z",
     "shell.execute_reply": "2022-09-26T01:28:20.167806Z",
     "shell.execute_reply.started": "2022-09-26T01:28:20.113684Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "from data import (\n",
    "    construct_anno_features,\n",
    "    construct_raw_mutation_features,\n",
    "    construct_raw_cnv_features,\n",
    "    canonicalize_mutations,\n",
    "    construct_anno_features_xon17\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e0b526a0-de1c-4cd1-807c-6546f95c019e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:21.159466Z",
     "iopub.status.busy": "2022-09-26T01:28:21.158790Z",
     "iopub.status.idle": "2022-09-26T01:28:21.225961Z",
     "shell.execute_reply": "2022-09-26T01:28:21.224401Z",
     "shell.execute_reply.started": "2022-09-26T01:28:21.159413Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>input</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>FBXW7 R505G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-4E-A92E</td>\n",
       "      <td>ARID1A S2269*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-4E-A92E</td>\n",
       "      <td>PIK3C2B I19I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-4E-A92E</td>\n",
       "      <td>CTNNB1 S37F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55794</th>\n",
       "      <td>TCGA-ZF-AA5N</td>\n",
       "      <td>ARID1A Q505Hfs*113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55795</th>\n",
       "      <td>TCGA-ZF-AA5N</td>\n",
       "      <td>HRAS K117N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55796</th>\n",
       "      <td>TCGA-ZF-AA5P</td>\n",
       "      <td>PPARG S249L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55797</th>\n",
       "      <td>TCGA-ZF-AA5P</td>\n",
       "      <td>TP53 H193R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55798</th>\n",
       "      <td>TCGA-ZF-AA5P</td>\n",
       "      <td>ETV4 R202W</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55799 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       submitter_id               input\n",
       "0      TCGA-2E-A9G8         FBXW7 R505G\n",
       "1      TCGA-2E-A9G8   TP53 E286_E287del\n",
       "2      TCGA-4E-A92E       ARID1A S2269*\n",
       "3      TCGA-4E-A92E        PIK3C2B I19I\n",
       "4      TCGA-4E-A92E         CTNNB1 S37F\n",
       "...             ...                 ...\n",
       "55794  TCGA-ZF-AA5N  ARID1A Q505Hfs*113\n",
       "55795  TCGA-ZF-AA5N          HRAS K117N\n",
       "55796  TCGA-ZF-AA5P         PPARG S249L\n",
       "55797  TCGA-ZF-AA5P          TP53 H193R\n",
       "55798  TCGA-ZF-AA5P          ETV4 R202W\n",
       "\n",
       "[55799 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tcga_mutations_df = pd.read_csv(\"../data/raw/tcga_point_mutations_incl_alive.csv\")\n",
    "tcga_mutations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1135ef82-2904-4cb9-a5db-631c3e96c700",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:22.339425Z",
     "iopub.status.busy": "2022-09-26T01:28:22.338599Z",
     "iopub.status.idle": "2022-09-26T01:28:22.368569Z",
     "shell.execute_reply": "2022-09-26T01:28:22.367597Z",
     "shell.execute_reply.started": "2022-09-26T01:28:22.339369Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>drug.name</th>\n",
       "      <th>response_description</th>\n",
       "      <th>response</th>\n",
       "      <th>drug_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-G2-A2EC</td>\n",
       "      <td>Methotrexate</td>\n",
       "      <td>Partial Response</td>\n",
       "      <td>1</td>\n",
       "      <td>METHOTREXATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-G2-A2EC</td>\n",
       "      <td>Doxorubicin</td>\n",
       "      <td>Partial Response</td>\n",
       "      <td>1</td>\n",
       "      <td>DOXORUBICIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-G2-A2EC</td>\n",
       "      <td>Vinblastine</td>\n",
       "      <td>Partial Response</td>\n",
       "      <td>1</td>\n",
       "      <td>VINBLASTINE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-G2-A2EC</td>\n",
       "      <td>Cisplatin</td>\n",
       "      <td>Partial Response</td>\n",
       "      <td>1</td>\n",
       "      <td>CISPLATIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-G2-A2EJ</td>\n",
       "      <td>Paclitaxel</td>\n",
       "      <td>Stable Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>PACLITAXEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1244</th>\n",
       "      <td>TCGA-BG-A0VZ</td>\n",
       "      <td>Cisplatin</td>\n",
       "      <td>Complete Response</td>\n",
       "      <td>1</td>\n",
       "      <td>CISPLATIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1245</th>\n",
       "      <td>TCGA-BG-A0VZ</td>\n",
       "      <td>Paclitaxel</td>\n",
       "      <td>Complete Response</td>\n",
       "      <td>1</td>\n",
       "      <td>PACLITAXEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246</th>\n",
       "      <td>TCGA-BG-A0VZ</td>\n",
       "      <td>Doxorubicin</td>\n",
       "      <td>Complete Response</td>\n",
       "      <td>1</td>\n",
       "      <td>DOXORUBICIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1247</th>\n",
       "      <td>TCGA-BG-A0VT</td>\n",
       "      <td>Carboplatin</td>\n",
       "      <td>Complete Response</td>\n",
       "      <td>1</td>\n",
       "      <td>CARBOPLATIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1248</th>\n",
       "      <td>TCGA-BG-A0VT</td>\n",
       "      <td>Paclitaxel</td>\n",
       "      <td>Complete Response</td>\n",
       "      <td>1</td>\n",
       "      <td>PACLITAXEL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1249 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      submitter_id     drug.name response_description  response     drug_name\n",
       "0     TCGA-G2-A2EC  Methotrexate     Partial Response         1  METHOTREXATE\n",
       "1     TCGA-G2-A2EC   Doxorubicin     Partial Response         1   DOXORUBICIN\n",
       "2     TCGA-G2-A2EC   Vinblastine     Partial Response         1   VINBLASTINE\n",
       "3     TCGA-G2-A2EC     Cisplatin     Partial Response         1     CISPLATIN\n",
       "4     TCGA-G2-A2EJ    Paclitaxel       Stable Disease         0    PACLITAXEL\n",
       "...            ...           ...                  ...       ...           ...\n",
       "1244  TCGA-BG-A0VZ     Cisplatin    Complete Response         1     CISPLATIN\n",
       "1245  TCGA-BG-A0VZ    Paclitaxel    Complete Response         1    PACLITAXEL\n",
       "1246  TCGA-BG-A0VZ   Doxorubicin    Complete Response         1   DOXORUBICIN\n",
       "1247  TCGA-BG-A0VT   Carboplatin    Complete Response         1   CARBOPLATIN\n",
       "1248  TCGA-BG-A0VT    Paclitaxel    Complete Response         1    PACLITAXEL\n",
       "\n",
       "[1249 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tcga_response = pd.read_csv(\"../data/processed/TCGA_drug_response_010222.csv\")\n",
    "tcga_response.rename(\n",
    "    columns={\n",
    "        \"patient.arr\": \"submitter_id\",\n",
    "        \"drug\": \"drug_name\",\n",
    "        \"response\": \"response_description\",\n",
    "        \"response_cat\": \"response\",\n",
    "    },\n",
    "    inplace=True,\n",
    ")\n",
    "tcga_response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "36a0a797-b0d2-42a7-9086-350d1d8333f5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:23.726042Z",
     "iopub.status.busy": "2022-09-26T01:28:23.725386Z",
     "iopub.status.idle": "2022-09-26T01:28:23.749234Z",
     "shell.execute_reply": "2022-09-26T01:28:23.747601Z",
     "shell.execute_reply.started": "2022-09-26T01:28:23.725988Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6121, 2)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tcga_mutations_df = tcga_mutations_df[tcga_mutations_df.submitter_id.isin(tcga_response.submitter_id.unique())].copy()\n",
    "tcga_mutations_df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a3f0c9d-9b9c-4dec-b14b-c6f822aa7e61",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-08-24T16:17:32.044263Z",
     "iopub.status.busy": "2022-08-24T16:17:32.043625Z",
     "iopub.status.idle": "2022-08-24T16:17:32.050209Z",
     "shell.execute_reply": "2022-08-24T16:17:32.048982Z",
     "shell.execute_reply.started": "2022-08-24T16:17:32.044209Z"
    },
    "tags": []
   },
   "source": [
    "## Variant Annotations with Annovar/Transvar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e31d3b84-4ff8-4e9e-a2db-3b0bf940278a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:27.779524Z",
     "iopub.status.busy": "2022-09-26T01:28:27.778708Z",
     "iopub.status.idle": "2022-09-26T01:28:27.790764Z",
     "shell.execute_reply": "2022-09-26T01:28:27.789278Z",
     "shell.execute_reply.started": "2022-09-26T01:28:27.779466Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "genes_324 = pd.read_csv(\"../../data/druid_1.3_data/gene2ind.txt\", header=None)[\n",
    "    0\n",
    "].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "86a5ec71-5769-4052-ae9b-6470fca12117",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:29.394130Z",
     "iopub.status.busy": "2022-09-26T01:28:29.393331Z",
     "iopub.status.idle": "2022-09-26T01:28:30.027802Z",
     "shell.execute_reply": "2022-09-26T01:28:30.026352Z",
     "shell.execute_reply.started": "2022-09-26T01:28:29.394075Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>point_mutation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>FBXW7 R505G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>AXL S447S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>BRD4 X1340_splice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>HRAS E31K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6116</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>RB1 S834*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6117</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>SMAD4 A406A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6118</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>TP53 A161T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6119</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>XPO1 F599F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6120</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6121 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      submitter_id     point_mutation\n",
       "0     TCGA-2E-A9G8        FBXW7 R505G\n",
       "1     TCGA-2E-A9G8  TP53 E286_E287del\n",
       "2     TCGA-A5-A1OH          AXL S447S\n",
       "3     TCGA-A5-A1OH  BRD4 X1340_splice\n",
       "4     TCGA-A5-A1OH          HRAS E31K\n",
       "...            ...                ...\n",
       "6116  TCGA-XF-AAN7          RB1 S834*\n",
       "6117  TCGA-XF-AAN7        SMAD4 A406A\n",
       "6118  TCGA-XF-AAN7         TP53 A161T\n",
       "6119  TCGA-XF-AAN7         XPO1 F599F\n",
       "6120  TCGA-XF-AAN7       ZNF217 L231V\n",
       "\n",
       "[6121 rows x 2 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract point mutations for the 324 F1 genes\n",
    "tcga_mutations = []\n",
    "for submitter_id in tcga_mutations_df.submitter_id.unique():\n",
    "    filtered_df = tcga_mutations_df[tcga_mutations_df.submitter_id == submitter_id]\n",
    "    point_mutations = filtered_df.input.values\n",
    "    point_mutations = canonicalize_mutations(point_mutations)\n",
    "    for gene in genes_324:\n",
    "        for point_mutation in point_mutations:\n",
    "            if f\"{gene} \" in point_mutation:\n",
    "                tcga_mutations.append((submitter_id, point_mutation.strip()))\n",
    "                \n",
    "tcga_mutations_filtered = pd.DataFrame(\n",
    "    tcga_mutations, columns=[\"submitter_id\", \"point_mutation\"]\n",
    ")\n",
    "tcga_mutations_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b8766abd-7fb8-4e93-8621-1335eb889370",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:31.028266Z",
     "iopub.status.busy": "2022-09-26T01:28:31.027396Z",
     "iopub.status.idle": "2022-09-26T01:28:32.553834Z",
     "shell.execute_reply": "2022-09-26T01:28:32.551267Z",
     "shell.execute_reply.started": "2022-09-26T01:28:31.028203Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# import dask\n",
    "\n",
    "# from tqdm import tqdm\n",
    "# from dask.distributed import Client\n",
    "\n",
    "\n",
    "# client = Client()\n",
    "# client.cluster.scale(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "658af8a2-d303-49a8-afef-dc9e956e3b89",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:28:32.558744Z",
     "iopub.status.busy": "2022-09-26T01:28:32.557590Z",
     "iopub.status.idle": "2022-09-26T01:28:32.588552Z",
     "shell.execute_reply": "2022-09-26T01:28:32.586845Z",
     "shell.execute_reply.started": "2022-09-26T01:28:32.558679Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# %load_ext jupyterlab_notify"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6fa61c09-e7b8-4d45-a719-d2e6158e9e8b",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "# %%notify\n",
    "\n",
    "# futures = []\n",
    "# for submitter_id in tcga_mutations_filtered.submitter_id.unique():\n",
    "#     filtered_df = tcga_mutations_filtered[\n",
    "#         tcga_mutations_filtered.submitter_id == submitter_id\n",
    "#     ]\n",
    "#     point_mutations = list(filtered_df.point_mutation.values)\n",
    "#     future = client.submit(construct_anno_features_xon17, submitter_id, point_mutations, False)\n",
    "#     futures.append(future)\n",
    "\n",
    "# anno_features_combined = client.gather(futures, errors=\"skip\")\n",
    "# client.shutdown()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "729172ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "# variant annotation on these point mutations\n",
    "import sys\n",
    "sys.path.append(\"../src/\")\n",
    "from data import (construct_anno_features, \n",
    "                  construct_raw_cnv_features, \n",
    "                  construct_raw_mutation_features, \n",
    "                  canonicalize_mutations,\n",
    "                  ALIAS_TO_CANONICAL_NAME_MAP, \n",
    "                  _is_valid_point_mutations,\n",
    "                  get_annotation_features,\n",
    "                  preprocess_annotation_features,\n",
    "                  MICEData\n",
    "                 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "d2332bac",
   "metadata": {},
   "outputs": [],
   "source": [
    "import logging\n",
    "import re\n",
    "import subprocess\n",
    "import tempfile\n",
    "import itertools\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "5c4658d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "REQUIRED_ANNOTATION_COLUMNS = [\n",
    "    \"SIFT_score\",\n",
    "    \"SIFT_converted_rankscore\",\n",
    "    \"SIFT_pred\",\n",
    "    \"SIFT4G_score\",\n",
    "    \"SIFT4G_converted_rankscore\",\n",
    "    \"SIFT4G_pred\",\n",
    "    \"LRT_score\",\n",
    "    \"LRT_converted_rankscore\",\n",
    "    \"LRT_pred\",\n",
    "    \"MutationTaster_score\",\n",
    "    \"MutationTaster_converted_rankscore\",\n",
    "    \"MutationTaster_pred\",\n",
    "    \"MutationAssessor_score\",\n",
    "    \"MutationAssessor_rankscore\",\n",
    "    \"MutationAssessor_pred\",\n",
    "    \"FATHMM_score\",\n",
    "    \"FATHMM_converted_rankscore\",\n",
    "    \"FATHMM_pred\",\n",
    "    \"PROVEAN_score\",\n",
    "    \"PROVEAN_converted_rankscore\",\n",
    "    \"PROVEAN_pred\",\n",
    "    \"MetaSVM_pred\",\n",
    "    \"M-CAP_score\",\n",
    "    \"M-CAP_rankscore\",\n",
    "    \"M-CAP_pred\",\n",
    "    \"MVP_score\",\n",
    "    \"MVP_rankscore\",\n",
    "    \"MPC_score\",\n",
    "    \"MPC_rankscore\",\n",
    "    \"PrimateAI_score\",\n",
    "    \"PrimateAI_rankscore\",\n",
    "    \"PrimateAI_pred\",\n",
    "    \"DEOGEN2_score\",\n",
    "    \"DEOGEN2_rankscore\",\n",
    "    \"DEOGEN2_pred\",\n",
    "    \"BayesDel_addAF_score\",\n",
    "    \"BayesDel_addAF_pred\",\n",
    "    \"BayesDel_noAF_score\",\n",
    "    \"BayesDel_noAF_rankscore\",\n",
    "    \"BayesDel_noAF_pred\",\n",
    "    \"ClinPred_score\",\n",
    "    \"ClinPred_rankscore\",\n",
    "    \"ClinPred_pred\",\n",
    "    \"LIST-S2_score\",\n",
    "    \"LIST-S2_rankscore\",\n",
    "    \"LIST-S2_pred\",\n",
    "    \"DANN_score\",\n",
    "    \"DANN_rankscore\",\n",
    "    \"fathmm-MKL_coding_score\",\n",
    "    \"fathmm-MKL_coding_rankscore\",\n",
    "    \"fathmm-MKL_coding_pred\",\n",
    "    \"fathmm-XF_coding_score\",\n",
    "    \"fathmm-XF_coding_rankscore\",\n",
    "    \"fathmm-XF_coding_pred\",\n",
    "    \"Eigen-raw_coding\",\n",
    "    \"Eigen-raw_coding_rankscore\",\n",
    "    \"Eigen-PC-raw_coding\",\n",
    "    \"Eigen-PC-raw_coding_rankscore\",\n",
    "]\n",
    "CATEGORICAL_COLUMNS = [\n",
    "    \"sift_pred\",\n",
    "    \"sift4g_pred\",\n",
    "    \"lrt_pred\",\n",
    "    \"mutationtaster_pred\",\n",
    "    \"mutationassessor_pred\",\n",
    "    \"fathmm_pred\",\n",
    "    \"provean_pred\",\n",
    "    \"metasvm_pred\",\n",
    "    \"m_cap_pred\",\n",
    "    \"primateai_pred\",\n",
    "    \"deogen2_pred\",\n",
    "    \"bayesdel_addaf_pred\",\n",
    "    \"bayesdel_noaf_pred\",\n",
    "    \"clinpred_pred\",\n",
    "    \"list_s2_pred\",\n",
    "    \"fathmm_mkl_coding_pred\",\n",
    "    \"fathmm_xf_coding_pred\",\n",
    "]\n",
    "\n",
    "# The thresholds used in PREDICTOR_LAMBDA_MAP are taken from the corresponding\n",
    "# technique's published paper/web page\n",
    "PREDICTOR_LAMBDA_MAP = {\n",
    "    \"sift_pred\": (\"sift_score\", lambda v: \"D\" if v <= 0.05 else \"T\"),\n",
    "    \"sift4g_pred\": (\"sift4g_score\", lambda v: \"D\" if v <= 0.05 else \"T\"),\n",
    "    \"lrt_pred\": (\"lrt_score\", lambda v: \"D\" if v <= 0.001 else \"U\"),\n",
    "    \"mutationtaster_pred\": (\n",
    "        \"mutationtaster_score\",\n",
    "        lambda v: None,\n",
    "    ),  # Threshold is not available and couldn't be derived from available values as well\n",
    "    \"mutationassessor_pred\": (\n",
    "        \"mutationassessor_score\",\n",
    "        lambda v: \"H\" if v >= 3.5 else (\"M\" if v >= 1.94 else \"L\"),\n",
    "    ),\n",
    "    \"fathmm_pred\": (\"fathmm_score\", lambda v: \"D\" if v < 1.5 else \"T\"),\n",
    "    \"provean_pred\": (\"provean_score\", lambda v: \"D\" if v <= 2.282 else \"N\"),\n",
    "    \"metasvm_pred\": (\n",
    "        \"metasvm_pred\",\n",
    "        lambda v: None,\n",
    "    ),  # No corresponding numeric score available for this method\n",
    "    \"m_cap_pred\": (\"m_cap_score\", lambda v: \"D\" if v >= 0.025 else \"T\"),\n",
    "    \"primateai_pred\": (\"primateai_score\", lambda v: \"D\" if v >= 0.803 else \"T\"),\n",
    "    \"deogen2_pred\": (\"deogen2_score\", lambda v: \"D\" if v >= 0.45 else \"T\"),\n",
    "    \"bayesdel_addaf_pred\": (\n",
    "        \"bayesdel_addaf_score\",\n",
    "        lambda v: \"D\" if v >= 0.0692 else \"T\",\n",
    "    ),\n",
    "    \"bayesdel_noaf_pred\": (\n",
    "        \"bayesdel_noaf_score\",\n",
    "        lambda v: \"D\" if v >= -0.0570 else \"T\",\n",
    "    ),\n",
    "    \"clinpred_pred\": (\"clinpred_score\", lambda v: \"D\" if v >= 0.5 else \"T\"),\n",
    "    \"list_s2_pred\": (\"list_s2_score\", lambda v: \"D\" if v >= 0.85 else \"T\"),\n",
    "    \"fathmm_mkl_coding_pred\": (\n",
    "        \"fathmm_mkl_coding_score\",\n",
    "        lambda v: \"D\" if v >= 0.5 else \"N\",\n",
    "    ),\n",
    "    \"fathmm_xf_coding_pred\": (\n",
    "        \"fathmm_xf_coding_score\",\n",
    "        lambda v: \"D\" if v >= 0.5 else \"N\",\n",
    "    ),\n",
    "}\n",
    "\n",
    "DELETERIOUS_VALUES = [\"D\", \"A\", \"H\", \"M\"]\n",
    "\n",
    "CNV_PATTERN = r\"loss|amplification\"\n",
    "\n",
    "GENES_324 = pd.read_csv(\"../../data/druid_1.3_data/gene2ind.txt\", header=None)[\n",
    "    0\n",
    "].tolist()\n",
    "\n",
    "ANNOTATION_SCRIPT_PATH = \"../script/goAAtoGv2.sh\"\n",
    "SPECIAL_CASES = r\"rearrangement|truncation|fs|del|ins\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "95666347",
   "metadata": {},
   "outputs": [],
   "source": [
    "def construct_anno_features_xon17_gpd(patient_id, patient_mutations, agg_features=False):\n",
    "    \"\"\"\n",
    "    TODO: Add support for other agg functions (mean, OR, etc) - as of 202209, only\n",
    "    sum is supported\n",
    "    Here, the aggregation is done as an average over all variants over all 17 algorithms.\n",
    "    \"\"\"\n",
    "    if agg_features:\n",
    "        logging.warn(\n",
    "            \"\"\"\n",
    "        Received agg_features=True -> As of now, construct_anno_features only supports sum aggregation.\n",
    "        Please ensure that the agg used in dataset definition is sum - if it is not sum, please pass\n",
    "        agg_features=False and perform agg in dataset definition\n",
    "        \"\"\"\n",
    "        )\n",
    "\n",
    "    if not _is_valid_point_mutations(patient_mutations):\n",
    "        return None\n",
    "\n",
    "    anno_features_combined_imputed_df = pd.read_csv(\n",
    "        \"../data/processed/anno_features_combined_imputed.csv\"\n",
    "    )\n",
    "    logging.info(anno_features_combined_imputed_df.shape)\n",
    "    anno_features_combined_imputed_df.set_index([\"input\"], inplace=True)\n",
    "    anno_features_combined_imputed_df.head()\n",
    "\n",
    "    canonical_mutations = canonicalize_mutations(patient_mutations)\n",
    "\n",
    "    mutations_with_missing_annotations = []\n",
    "    available_mutations = []\n",
    "    for mutation in canonical_mutations:\n",
    "        if mutation in anno_features_combined_imputed_df.index:\n",
    "            available_mutations.append(mutation)\n",
    "        elif not re.search(CNV_PATTERN, mutation, re.IGNORECASE):\n",
    "            mutations_with_missing_annotations.append(mutation)\n",
    "\n",
    "    if available_mutations:\n",
    "        patient_anno_features = anno_features_combined_imputed_df.loc[\n",
    "            available_mutations\n",
    "        ]\n",
    "        patient_anno_features = patient_anno_features[CATEGORICAL_COLUMNS].copy()\n",
    "    else:\n",
    "        patient_anno_features = None\n",
    "\n",
    "    if len(mutations_with_missing_annotations) != 0:\n",
    "        logging.info(\n",
    "            f\"Found mutations with missing annotations - {mutations_with_missing_annotations}\"\n",
    "        )\n",
    "        missing_annotations = get_annotation_features(\n",
    "            mutations_with_missing_annotations\n",
    "        )\n",
    "        if missing_annotations is not None:\n",
    "            missing_annotations = missing_annotations[\n",
    "                REQUIRED_ANNOTATION_COLUMNS\n",
    "            ].copy()\n",
    "            missing_annotations.reset_index(inplace=True)\n",
    "            missing_annotations = preprocess_annotation_features(missing_annotations)\n",
    "            missing_annotations = missing_annotations[~missing_annotations.duplicated()]\n",
    "            missing_annotations.reset_index(drop=True, inplace=True)\n",
    "            missing_annotations.set_index(\"input\", inplace=True)\n",
    "\n",
    "            numeric_columns = list(\n",
    "                column\n",
    "                for column in missing_annotations.columns\n",
    "                if pd.api.types.is_numeric_dtype(missing_annotations[column])\n",
    "            )\n",
    "            # Prepare mask by identifying rows that have all na values for numeric_columns\n",
    "            na_mask = None\n",
    "            for col in numeric_columns:\n",
    "                if type(na_mask) == pd.Series:\n",
    "                    na_mask = na_mask & missing_annotations[col].isna()\n",
    "                else:\n",
    "                    na_mask = missing_annotations[col].isna()\n",
    "\n",
    "            missing_annotations = missing_annotations[~na_mask]\n",
    "            numeric_df = missing_annotations[numeric_columns].copy()\n",
    "            logging.info(numeric_df.shape)\n",
    "            numeric_df.head()\n",
    "            numeric_df = pd.concat(\n",
    "                [numeric_df, anno_features_combined_imputed_df[numeric_columns]],\n",
    "            )\n",
    "\n",
    "            categorical_columns = [\n",
    "                column\n",
    "                for column in missing_annotations.columns\n",
    "                if column not in numeric_columns\n",
    "            ]\n",
    "            categorical_missing_annotations = missing_annotations[\n",
    "                categorical_columns\n",
    "            ].copy()\n",
    "            logging.info(categorical_missing_annotations.shape)\n",
    "            categorical_missing_annotations.head()\n",
    "\n",
    "            imp = MICEData(numeric_df)\n",
    "            # Impute missing values in numeric columns - Expensive!!\n",
    "            imp.update_all()\n",
    "            imputed_df = imp.data\n",
    "            assert numeric_df.shape == imputed_df.shape\n",
    "            imputed_df.index = numeric_df.index\n",
    "            imputed_df = imputed_df[\n",
    "                imputed_df.index.isin(mutations_with_missing_annotations)\n",
    "            ].copy()\n",
    "            numeric_imputed_df = pd.concat(\n",
    "                [categorical_missing_annotations, imputed_df,], axis=1,\n",
    "            )\n",
    "            logging.info(numeric_imputed_df.shape)\n",
    "            for column in CATEGORICAL_COLUMNS:\n",
    "                logging.info(\n",
    "                    column,\n",
    "                    numeric_imputed_df[column].unique(),\n",
    "                    len(numeric_imputed_df[numeric_imputed_df[column].isna()]),\n",
    "                )\n",
    "                col_na_mask = numeric_imputed_df[column].isna()\n",
    "                numeric_imputed_df.loc[col_na_mask, column] = numeric_imputed_df[\n",
    "                    col_na_mask\n",
    "                ][PREDICTOR_LAMBDA_MAP[column][0]].apply(\n",
    "                    PREDICTOR_LAMBDA_MAP[column][1]\n",
    "                )\n",
    "\n",
    "                # logging.info(\n",
    "                #     column,\n",
    "                #     numeric_imputed_df[column].unique(),\n",
    "                #     len(numeric_imputed_df[numeric_imputed_df[column].isna()]),\n",
    "                # )\n",
    "\n",
    "            numeric_imputed_df = numeric_imputed_df.dropna()\n",
    "            logging.info(numeric_imputed_df.shape)\n",
    "            missing_anno_features_df = numeric_imputed_df[CATEGORICAL_COLUMNS].copy()\n",
    "\n",
    "            patient_anno_features = pd.concat(\n",
    "                [missing_anno_features_df, patient_anno_features]\n",
    "            )\n",
    "            \n",
    "    if patient_anno_features is not None:\n",
    "        for col in patient_anno_features.columns:\n",
    "            patient_anno_features[col] = patient_anno_features[col].apply(\n",
    "                lambda v: 1 if v in DELETERIOUS_VALUES else 0\n",
    "            )\n",
    "\n",
    "        patient_anno_features.reset_index(inplace=True)\n",
    "        patient_anno_features[\"gene\"] = patient_anno_features.input.apply(\n",
    "            lambda gene_mut: gene_mut.split(\" \")[0]\n",
    "        )\n",
    "    return patient_anno_features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d97353c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>point_mutation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>FBXW7 R505G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>AXL S447S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>BRD4 X1340_splice</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>HRAS E31K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6116</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>RB1 S834*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6117</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>SMAD4 A406A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6118</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>TP53 A161T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6119</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>XPO1 F599F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6120</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6121 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      submitter_id     point_mutation\n",
       "0     TCGA-2E-A9G8        FBXW7 R505G\n",
       "1     TCGA-2E-A9G8  TP53 E286_E287del\n",
       "2     TCGA-A5-A1OH          AXL S447S\n",
       "3     TCGA-A5-A1OH  BRD4 X1340_splice\n",
       "4     TCGA-A5-A1OH          HRAS E31K\n",
       "...            ...                ...\n",
       "6116  TCGA-XF-AAN7          RB1 S834*\n",
       "6117  TCGA-XF-AAN7        SMAD4 A406A\n",
       "6118  TCGA-XF-AAN7         TP53 A161T\n",
       "6119  TCGA-XF-AAN7         XPO1 F599F\n",
       "6120  TCGA-XF-AAN7       ZNF217 L231V\n",
       "\n",
       "[6121 rows x 2 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tcga_mutations_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b4fc7788",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "# %%notify\n",
    "\n",
    "# futures = []\n",
    "anno_features_combined = pd.DataFrame()\n",
    "for patient_id in tcga_mutations_filtered.submitter_id.unique():\n",
    "    filtered_df = tcga_mutations_filtered[\n",
    "        tcga_mutations_filtered.submitter_id == submitter_id\n",
    "    ]\n",
    "    point_mutations = list(filtered_df.point_mutation.values)\n",
    "    res = construct_anno_features_xon17_gpd(patient_id, point_mutations, False)\n",
    "    anno_features_combined = pd.concat([anno_features_combined, res], ignore_index = True)\n",
    "#     future = client.submit(construct_anno_features_xon17_gpd, patient_id, point_mutations, False)\n",
    "#     futures.append(future)\n",
    "\n",
    "# anno_features_combined = client.gather(futures, errors=\"skip\")\n",
    "# client.shutdown()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "1c13cadc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>primateai_pred</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATRX E733K</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>ATRX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CTCF E376*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTCF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KRAS G12D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NOTCH1 V2119M</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NOTCH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RB1 S834*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4167</th>\n",
       "      <td>KRAS G12D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4168</th>\n",
       "      <td>NOTCH1 V2119M</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NOTCH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4169</th>\n",
       "      <td>RB1 S834*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4170</th>\n",
       "      <td>TP53 A161T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4171</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4172 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              input  sift_pred  sift4g_pred  lrt_pred  mutationtaster_pred  \\\n",
       "0        ATRX E733K          1            1         1                    1   \n",
       "1        CTCF E376*          0            0         1                    1   \n",
       "2         KRAS G12D          1            1         1                    1   \n",
       "3     NOTCH1 V2119M          1            0         0                    1   \n",
       "4         RB1 S834*          0            0         1                    1   \n",
       "...             ...        ...          ...       ...                  ...   \n",
       "4167      KRAS G12D          1            1         1                    1   \n",
       "4168  NOTCH1 V2119M          1            0         0                    1   \n",
       "4169      RB1 S834*          0            0         1                    1   \n",
       "4170     TP53 A161T          1            1         1                    1   \n",
       "4171   ZNF217 L231V          1            1         1                    1   \n",
       "\n",
       "      mutationassessor_pred  fathmm_pred  provean_pred  metasvm_pred  \\\n",
       "0                         0            1             0             1   \n",
       "1                         0            0             0             0   \n",
       "2                         1            0             1             1   \n",
       "3                         0            1             0             1   \n",
       "4                         0            0             0             0   \n",
       "...                     ...          ...           ...           ...   \n",
       "4167                      1            0             1             1   \n",
       "4168                      0            1             0             1   \n",
       "4169                      0            0             0             0   \n",
       "4170                      1            1             1             1   \n",
       "4171                      1            0             1             0   \n",
       "\n",
       "      m_cap_pred  primateai_pred  deogen2_pred  bayesdel_addaf_pred  \\\n",
       "0              1               0             0                    1   \n",
       "1              0               0             0                    1   \n",
       "2              1               1             1                    1   \n",
       "3              1               0             0                    0   \n",
       "4              0               0             0                    1   \n",
       "...          ...             ...           ...                  ...   \n",
       "4167           1               1             1                    1   \n",
       "4168           1               0             0                    0   \n",
       "4169           0               0             0                    1   \n",
       "4170           1               0             1                    1   \n",
       "4171           1               0             1                    1   \n",
       "\n",
       "      bayesdel_noaf_pred  clinpred_pred  list_s2_pred  fathmm_mkl_coding_pred  \\\n",
       "0                      1              1             1                       1   \n",
       "1                      1              0             0                       1   \n",
       "2                      1              1             0                       1   \n",
       "3                      0              1             1                       1   \n",
       "4                      1              0             0                       1   \n",
       "...                  ...            ...           ...                     ...   \n",
       "4167                   1              1             0                       1   \n",
       "4168                   0              1             1                       1   \n",
       "4169                   1              0             0                       1   \n",
       "4170                   1              1             1                       1   \n",
       "4171                   0              1             1                       1   \n",
       "\n",
       "      fathmm_xf_coding_pred    gene  \n",
       "0                         0    ATRX  \n",
       "1                         0    CTCF  \n",
       "2                         1    KRAS  \n",
       "3                         1  NOTCH1  \n",
       "4                         0     RB1  \n",
       "...                     ...     ...  \n",
       "4167                      1    KRAS  \n",
       "4168                      1  NOTCH1  \n",
       "4169                      0     RB1  \n",
       "4170                      1    TP53  \n",
       "4171                      0  ZNF217  \n",
       "\n",
       "[4172 rows x 19 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_features_combined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3c1e4366-6b06-422b-9c22-130303ddb982",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-26T01:37:19.119095Z",
     "iopub.status.busy": "2022-09-26T01:37:19.118753Z",
     "iopub.status.idle": "2022-09-26T01:37:19.615142Z",
     "shell.execute_reply": "2022-09-26T01:37:19.614188Z",
     "shell.execute_reply.started": "2022-09-26T01:37:19.119073Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>primateai_pred</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>input</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ATRX E733K</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>ATRX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTCF E376*</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>CTCF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KRAS G12D</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH1 V2119M</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NOTCH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RB1 S834*</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KRAS G12D</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH1 V2119M</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NOTCH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RB1 S834*</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53 A161T</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF217 L231V</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4172 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               sift_pred  sift4g_pred  lrt_pred  mutationtaster_pred  \\\n",
       "input                                                                  \n",
       "ATRX E733K             1            1         1                    1   \n",
       "CTCF E376*             0            0         1                    1   \n",
       "KRAS G12D              1            1         1                    1   \n",
       "NOTCH1 V2119M          1            0         0                    1   \n",
       "RB1 S834*              0            0         1                    1   \n",
       "...                  ...          ...       ...                  ...   \n",
       "KRAS G12D              1            1         1                    1   \n",
       "NOTCH1 V2119M          1            0         0                    1   \n",
       "RB1 S834*              0            0         1                    1   \n",
       "TP53 A161T             1            1         1                    1   \n",
       "ZNF217 L231V           1            1         1                    1   \n",
       "\n",
       "               mutationassessor_pred  fathmm_pred  provean_pred  metasvm_pred  \\\n",
       "input                                                                           \n",
       "ATRX E733K                         0            1             0             1   \n",
       "CTCF E376*                         0            0             0             0   \n",
       "KRAS G12D                          1            0             1             1   \n",
       "NOTCH1 V2119M                      0            1             0             1   \n",
       "RB1 S834*                          0            0             0             0   \n",
       "...                              ...          ...           ...           ...   \n",
       "KRAS G12D                          1            0             1             1   \n",
       "NOTCH1 V2119M                      0            1             0             1   \n",
       "RB1 S834*                          0            0             0             0   \n",
       "TP53 A161T                         1            1             1             1   \n",
       "ZNF217 L231V                       1            0             1             0   \n",
       "\n",
       "               m_cap_pred  primateai_pred  deogen2_pred  bayesdel_addaf_pred  \\\n",
       "input                                                                          \n",
       "ATRX E733K              1               0             0                    1   \n",
       "CTCF E376*              0               0             0                    1   \n",
       "KRAS G12D               1               1             1                    1   \n",
       "NOTCH1 V2119M           1               0             0                    0   \n",
       "RB1 S834*               0               0             0                    1   \n",
       "...                   ...             ...           ...                  ...   \n",
       "KRAS G12D               1               1             1                    1   \n",
       "NOTCH1 V2119M           1               0             0                    0   \n",
       "RB1 S834*               0               0             0                    1   \n",
       "TP53 A161T              1               0             1                    1   \n",
       "ZNF217 L231V            1               0             1                    1   \n",
       "\n",
       "               bayesdel_noaf_pred  clinpred_pred  list_s2_pred  \\\n",
       "input                                                            \n",
       "ATRX E733K                      1              1             1   \n",
       "CTCF E376*                      1              0             0   \n",
       "KRAS G12D                       1              1             0   \n",
       "NOTCH1 V2119M                   0              1             1   \n",
       "RB1 S834*                       1              0             0   \n",
       "...                           ...            ...           ...   \n",
       "KRAS G12D                       1              1             0   \n",
       "NOTCH1 V2119M                   0              1             1   \n",
       "RB1 S834*                       1              0             0   \n",
       "TP53 A161T                      1              1             1   \n",
       "ZNF217 L231V                    0              1             1   \n",
       "\n",
       "               fathmm_mkl_coding_pred  fathmm_xf_coding_pred    gene  \n",
       "input                                                                 \n",
       "ATRX E733K                          1                      0    ATRX  \n",
       "CTCF E376*                          1                      0    CTCF  \n",
       "KRAS G12D                           1                      1    KRAS  \n",
       "NOTCH1 V2119M                       1                      1  NOTCH1  \n",
       "RB1 S834*                           1                      0     RB1  \n",
       "...                               ...                    ...     ...  \n",
       "KRAS G12D                           1                      1    KRAS  \n",
       "NOTCH1 V2119M                       1                      1  NOTCH1  \n",
       "RB1 S834*                           1                      0     RB1  \n",
       "TP53 A161T                          1                      1    TP53  \n",
       "ZNF217 L231V                        1                      0  ZNF217  \n",
       "\n",
       "[4172 rows x 18 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agg_anno_features_combined_df = anno_features_combined#pd.concat(anno_features_combined).reset_index()\n",
    "agg_anno_features_combined_df.set_index(\"input\", inplace=True)\n",
    "agg_anno_features_combined_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "208f9689",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>point_mutation</th>\n",
       "      <th>input</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>...</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>primateai_pred</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>FBXW7 R505G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>AXL S447S</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>BRD4 X1340_splice</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>HRAS E31K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16826</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16827</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16828</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16829</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16830</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16831 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       submitter_id     point_mutation         input  sift_pred  sift4g_pred  \\\n",
       "0      TCGA-2E-A9G8        FBXW7 R505G           NaN        NaN          NaN   \n",
       "1      TCGA-2E-A9G8  TP53 E286_E287del           NaN        NaN          NaN   \n",
       "2      TCGA-A5-A1OH          AXL S447S           NaN        NaN          NaN   \n",
       "3      TCGA-A5-A1OH  BRD4 X1340_splice           NaN        NaN          NaN   \n",
       "4      TCGA-A5-A1OH          HRAS E31K           NaN        NaN          NaN   \n",
       "...             ...                ...           ...        ...          ...   \n",
       "16826  TCGA-XF-AAN7       ZNF217 L231V  ZNF217 L231V        1.0          1.0   \n",
       "16827  TCGA-XF-AAN7       ZNF217 L231V  ZNF217 L231V        1.0          1.0   \n",
       "16828  TCGA-XF-AAN7       ZNF217 L231V  ZNF217 L231V        1.0          1.0   \n",
       "16829  TCGA-XF-AAN7       ZNF217 L231V  ZNF217 L231V        1.0          1.0   \n",
       "16830  TCGA-XF-AAN7       ZNF217 L231V  ZNF217 L231V        1.0          1.0   \n",
       "\n",
       "       lrt_pred  mutationtaster_pred  mutationassessor_pred  fathmm_pred  \\\n",
       "0           NaN                  NaN                    NaN          NaN   \n",
       "1           NaN                  NaN                    NaN          NaN   \n",
       "2           NaN                  NaN                    NaN          NaN   \n",
       "3           NaN                  NaN                    NaN          NaN   \n",
       "4           NaN                  NaN                    NaN          NaN   \n",
       "...         ...                  ...                    ...          ...   \n",
       "16826       1.0                  1.0                    1.0          0.0   \n",
       "16827       1.0                  1.0                    1.0          0.0   \n",
       "16828       1.0                  1.0                    1.0          0.0   \n",
       "16829       1.0                  1.0                    1.0          0.0   \n",
       "16830       1.0                  1.0                    1.0          0.0   \n",
       "\n",
       "       provean_pred  ...  m_cap_pred  primateai_pred  deogen2_pred  \\\n",
       "0               NaN  ...         NaN             NaN           NaN   \n",
       "1               NaN  ...         NaN             NaN           NaN   \n",
       "2               NaN  ...         NaN             NaN           NaN   \n",
       "3               NaN  ...         NaN             NaN           NaN   \n",
       "4               NaN  ...         NaN             NaN           NaN   \n",
       "...             ...  ...         ...             ...           ...   \n",
       "16826           1.0  ...         1.0             0.0           1.0   \n",
       "16827           1.0  ...         1.0             0.0           1.0   \n",
       "16828           1.0  ...         1.0             0.0           1.0   \n",
       "16829           1.0  ...         1.0             0.0           1.0   \n",
       "16830           1.0  ...         1.0             0.0           1.0   \n",
       "\n",
       "       bayesdel_addaf_pred  bayesdel_noaf_pred  clinpred_pred  list_s2_pred  \\\n",
       "0                      NaN                 NaN            NaN           NaN   \n",
       "1                      NaN                 NaN            NaN           NaN   \n",
       "2                      NaN                 NaN            NaN           NaN   \n",
       "3                      NaN                 NaN            NaN           NaN   \n",
       "4                      NaN                 NaN            NaN           NaN   \n",
       "...                    ...                 ...            ...           ...   \n",
       "16826                  1.0                 0.0            1.0           1.0   \n",
       "16827                  1.0                 0.0            1.0           1.0   \n",
       "16828                  1.0                 0.0            1.0           1.0   \n",
       "16829                  1.0                 0.0            1.0           1.0   \n",
       "16830                  1.0                 0.0            1.0           1.0   \n",
       "\n",
       "       fathmm_mkl_coding_pred  fathmm_xf_coding_pred    gene  \n",
       "0                         NaN                    NaN     NaN  \n",
       "1                         NaN                    NaN     NaN  \n",
       "2                         NaN                    NaN     NaN  \n",
       "3                         NaN                    NaN     NaN  \n",
       "4                         NaN                    NaN     NaN  \n",
       "...                       ...                    ...     ...  \n",
       "16826                     1.0                    0.0  ZNF217  \n",
       "16827                     1.0                    0.0  ZNF217  \n",
       "16828                     1.0                    0.0  ZNF217  \n",
       "16829                     1.0                    0.0  ZNF217  \n",
       "16830                     1.0                    0.0  ZNF217  \n",
       "\n",
       "[16831 rows x 21 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# HERE!! - Now merge with original and save\n",
    "merged_1 = pd.merge(tcga_mutations_filtered, agg_anno_features_combined_df.reset_index(), left_on = \"point_mutation\", right_on=\"input\", how=\"left\")\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "667fe6f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "submitter_id                 0\n",
       "point_mutation               0\n",
       "sift_pred                 6103\n",
       "sift4g_pred               6103\n",
       "lrt_pred                  6103\n",
       "mutationtaster_pred       6103\n",
       "mutationassessor_pred     6103\n",
       "fathmm_pred               6103\n",
       "provean_pred              6103\n",
       "metasvm_pred              6103\n",
       "m_cap_pred                6103\n",
       "primateai_pred            6103\n",
       "deogen2_pred              6103\n",
       "bayesdel_addaf_pred       6103\n",
       "bayesdel_noaf_pred        6103\n",
       "clinpred_pred             6103\n",
       "list_s2_pred              6103\n",
       "fathmm_mkl_coding_pred    6103\n",
       "fathmm_xf_coding_pred     6103\n",
       "gene                      6103\n",
       "dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1.drop(\"input\", axis = 1, inplace=True)\n",
    "merged_1.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6fc963f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For those point mutations without annotations\n",
    "merged_1.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a0c44bae",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_1[\"gene\"] = merged_1[\"point_mutation\"].apply(lambda x: x.split(\" \")[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "5c8673c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KRAS      6011\n",
       "TP53      1507\n",
       "RB1       1240\n",
       "ATRX       642\n",
       "NOTCH1     640\n",
       "          ... \n",
       "BCL2L2       2\n",
       "RAD51D       1\n",
       "AURKA        1\n",
       "CEBPA        1\n",
       "CCND3        1\n",
       "Name: gene, Length: 312, dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1[\"gene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "f8de2eb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>point_mutation</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>...</th>\n",
       "      <th>primateai_pred</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "      <th>1plusxon17_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>FBXW7 R505G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-2E-A9G8</td>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>AXL S447S</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AXL</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>BRD4 X1340_splice</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-A5-A1OH</td>\n",
       "      <td>HRAS E31K</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16826</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16827</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16828</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16829</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16830</th>\n",
       "      <td>TCGA-XF-AAN7</td>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16831 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       submitter_id     point_mutation  sift_pred  sift4g_pred  lrt_pred  \\\n",
       "0      TCGA-2E-A9G8        FBXW7 R505G        0.0          0.0       0.0   \n",
       "1      TCGA-2E-A9G8  TP53 E286_E287del        0.0          0.0       0.0   \n",
       "2      TCGA-A5-A1OH          AXL S447S        0.0          0.0       0.0   \n",
       "3      TCGA-A5-A1OH  BRD4 X1340_splice        0.0          0.0       0.0   \n",
       "4      TCGA-A5-A1OH          HRAS E31K        0.0          0.0       0.0   \n",
       "...             ...                ...        ...          ...       ...   \n",
       "16826  TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "16827  TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "16828  TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "16829  TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "16830  TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "\n",
       "       mutationtaster_pred  mutationassessor_pred  fathmm_pred  provean_pred  \\\n",
       "0                      0.0                    0.0          0.0           0.0   \n",
       "1                      0.0                    0.0          0.0           0.0   \n",
       "2                      0.0                    0.0          0.0           0.0   \n",
       "3                      0.0                    0.0          0.0           0.0   \n",
       "4                      0.0                    0.0          0.0           0.0   \n",
       "...                    ...                    ...          ...           ...   \n",
       "16826                  1.0                    1.0          0.0           1.0   \n",
       "16827                  1.0                    1.0          0.0           1.0   \n",
       "16828                  1.0                    1.0          0.0           1.0   \n",
       "16829                  1.0                    1.0          0.0           1.0   \n",
       "16830                  1.0                    1.0          0.0           1.0   \n",
       "\n",
       "       metasvm_pred  ...  primateai_pred  deogen2_pred  bayesdel_addaf_pred  \\\n",
       "0               0.0  ...             0.0           0.0                  0.0   \n",
       "1               0.0  ...             0.0           0.0                  0.0   \n",
       "2               0.0  ...             0.0           0.0                  0.0   \n",
       "3               0.0  ...             0.0           0.0                  0.0   \n",
       "4               0.0  ...             0.0           0.0                  0.0   \n",
       "...             ...  ...             ...           ...                  ...   \n",
       "16826           0.0  ...             0.0           1.0                  1.0   \n",
       "16827           0.0  ...             0.0           1.0                  1.0   \n",
       "16828           0.0  ...             0.0           1.0                  1.0   \n",
       "16829           0.0  ...             0.0           1.0                  1.0   \n",
       "16830           0.0  ...             0.0           1.0                  1.0   \n",
       "\n",
       "       bayesdel_noaf_pred  clinpred_pred  list_s2_pred  \\\n",
       "0                     0.0            0.0           0.0   \n",
       "1                     0.0            0.0           0.0   \n",
       "2                     0.0            0.0           0.0   \n",
       "3                     0.0            0.0           0.0   \n",
       "4                     0.0            0.0           0.0   \n",
       "...                   ...            ...           ...   \n",
       "16826                 0.0            1.0           1.0   \n",
       "16827                 0.0            1.0           1.0   \n",
       "16828                 0.0            1.0           1.0   \n",
       "16829                 0.0            1.0           1.0   \n",
       "16830                 0.0            1.0           1.0   \n",
       "\n",
       "       fathmm_mkl_coding_pred  fathmm_xf_coding_pred    gene 1plusxon17_score  \n",
       "0                         0.0                    0.0   FBXW7         1.000000  \n",
       "1                         0.0                    0.0    TP53         1.000000  \n",
       "2                         0.0                    0.0     AXL         1.000000  \n",
       "3                         0.0                    0.0    BRD4         1.000000  \n",
       "4                         0.0                    0.0    HRAS         1.000000  \n",
       "...                       ...                    ...     ...              ...  \n",
       "16826                     1.0                    0.0  ZNF217         1.705882  \n",
       "16827                     1.0                    0.0  ZNF217         1.705882  \n",
       "16828                     1.0                    0.0  ZNF217         1.705882  \n",
       "16829                     1.0                    0.0  ZNF217         1.705882  \n",
       "16830                     1.0                    0.0  ZNF217         1.705882  \n",
       "\n",
       "[16831 rows x 21 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1[\"1plusxon17_score\"] = 1 + merged_1[CATEGORICAL_COLUMNS].sum(axis=1)/17\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0973cb8e",
   "metadata": {},
   "source": [
    "### Output: Xon17 annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "7e647050",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>gene</th>\n",
       "      <th>ABL1</th>\n",
       "      <th>ACVR1B</th>\n",
       "      <th>AKT1</th>\n",
       "      <th>AKT2</th>\n",
       "      <th>AKT3</th>\n",
       "      <th>ALK</th>\n",
       "      <th>ALOX12B</th>\n",
       "      <th>APC</th>\n",
       "      <th>AR</th>\n",
       "      <th>ARAF</th>\n",
       "      <th>...</th>\n",
       "      <th>TYRO3</th>\n",
       "      <th>VEGFA</th>\n",
       "      <th>VHL</th>\n",
       "      <th>WHSC1</th>\n",
       "      <th>WHSC1L1</th>\n",
       "      <th>WT1</th>\n",
       "      <th>XPO1</th>\n",
       "      <th>XRCC2</th>\n",
       "      <th>ZNF217</th>\n",
       "      <th>ZNF703</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4384</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4390</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4398</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4402</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4427</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XX-A899</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>596 rows × 312 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "gene          ABL1  ACVR1B  AKT1  AKT2  AKT3  ALK  ALOX12B  APC   AR  ARAF  \\\n",
       "submitter_id                                                                 \n",
       "TCGA-05-4384   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4390   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4398   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4402   0.0     0.0   0.0   0.0   0.0  0.0      0.0  1.0  0.0   0.0   \n",
       "TCGA-05-4427   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "...            ...     ...   ...   ...   ...  ...      ...  ...  ...   ...   \n",
       "TCGA-XF-AAN5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XF-AAN7   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XX-A899   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-Z7-A8R5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-Z7-A8R6   0.0     0.0   0.0   0.0   0.0  1.0      0.0  0.0  0.0   0.0   \n",
       "\n",
       "gene          ...  TYRO3  VEGFA  VHL  WHSC1  WHSC1L1  WT1  XPO1  XRCC2  \\\n",
       "submitter_id  ...                                                        \n",
       "TCGA-05-4384  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4390  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4398  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4402  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4427  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "...           ...    ...    ...  ...    ...      ...  ...   ...    ...   \n",
       "TCGA-XF-AAN5  ...    0.0    0.0  0.0    1.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-XF-AAN7  ...    0.0    0.0  0.0    0.0      0.0  0.0   1.0    0.0   \n",
       "TCGA-XX-A899  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-Z7-A8R5  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-Z7-A8R6  ...    0.0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "\n",
       "gene            ZNF217  ZNF703  \n",
       "submitter_id                    \n",
       "TCGA-05-4384  0.000000     0.0  \n",
       "TCGA-05-4390  0.000000     0.0  \n",
       "TCGA-05-4398  0.000000     0.0  \n",
       "TCGA-05-4402  0.000000     0.0  \n",
       "TCGA-05-4427  0.000000     0.0  \n",
       "...                ...     ...  \n",
       "TCGA-XF-AAN5  0.000000     0.0  \n",
       "TCGA-XF-AAN7  1.705882     0.0  \n",
       "TCGA-XX-A899  0.000000     0.0  \n",
       "TCGA-Z7-A8R5  0.000000     0.0  \n",
       "TCGA-Z7-A8R6  0.000000     0.0  \n",
       "\n",
       "[596 rows x 312 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_gene_matrix_xon17 = merged_1.pivot_table(index=\"submitter_id\", columns=\"gene\", values=\"1plusxon17_score\", aggfunc=\"max\")\n",
    "patient_gene_matrix_xon17.fillna(0, inplace=True)\n",
    "patient_gene_matrix_xon17"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "33f8bf0d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(596, 324)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for g in GENES_324:\n",
    "    if g not in patient_gene_matrix_xon17.columns:\n",
    "        patient_gene_matrix_xon17[g] = 0\n",
    "patient_gene_matrix_xon17.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "3c0087eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "patient_gene_matrix_xon17[GENES_324].reset_index().to_csv(\"../data/processed/tcga_anno_features_xon17.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "637e1074",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ABL1</th>\n",
       "      <th>ACVR1B</th>\n",
       "      <th>AKT1</th>\n",
       "      <th>AKT2</th>\n",
       "      <th>AKT3</th>\n",
       "      <th>ALK</th>\n",
       "      <th>ALOX12B</th>\n",
       "      <th>APC</th>\n",
       "      <th>AR</th>\n",
       "      <th>ARAF</th>\n",
       "      <th>...</th>\n",
       "      <th>U2AF1</th>\n",
       "      <th>VEGFA</th>\n",
       "      <th>VHL</th>\n",
       "      <th>WHSC1</th>\n",
       "      <th>WHSC1L1</th>\n",
       "      <th>WT1</th>\n",
       "      <th>XPO1</th>\n",
       "      <th>XRCC2</th>\n",
       "      <th>ZNF217</th>\n",
       "      <th>ZNF703</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4384</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4390</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4398</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4402</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4427</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XX-A899</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>596 rows × 324 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              ABL1  ACVR1B  AKT1  AKT2  AKT3  ALK  ALOX12B  APC   AR  ARAF  \\\n",
       "submitter_id                                                                 \n",
       "TCGA-05-4384   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4390   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4398   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4402   0.0     0.0   0.0   0.0   0.0  0.0      0.0  1.0  0.0   0.0   \n",
       "TCGA-05-4427   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "...            ...     ...   ...   ...   ...  ...      ...  ...  ...   ...   \n",
       "TCGA-XF-AAN5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XF-AAN7   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XX-A899   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-Z7-A8R5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-Z7-A8R6   0.0     0.0   0.0   0.0   0.0  1.0      0.0  0.0  0.0   0.0   \n",
       "\n",
       "              ...  U2AF1  VEGFA  VHL  WHSC1  WHSC1L1  WT1  XPO1  XRCC2  \\\n",
       "submitter_id  ...                                                        \n",
       "TCGA-05-4384  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4390  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4398  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4402  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-05-4427  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "...           ...    ...    ...  ...    ...      ...  ...   ...    ...   \n",
       "TCGA-XF-AAN5  ...      0    0.0  0.0    1.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-XF-AAN7  ...      0    0.0  0.0    0.0      0.0  0.0   1.0    0.0   \n",
       "TCGA-XX-A899  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-Z7-A8R5  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "TCGA-Z7-A8R6  ...      0    0.0  0.0    0.0      0.0  0.0   0.0    0.0   \n",
       "\n",
       "                ZNF217  ZNF703  \n",
       "submitter_id                    \n",
       "TCGA-05-4384  0.000000     0.0  \n",
       "TCGA-05-4390  0.000000     0.0  \n",
       "TCGA-05-4398  0.000000     0.0  \n",
       "TCGA-05-4402  0.000000     0.0  \n",
       "TCGA-05-4427  0.000000     0.0  \n",
       "...                ...     ...  \n",
       "TCGA-XF-AAN5  0.000000     0.0  \n",
       "TCGA-XF-AAN7  1.705882     0.0  \n",
       "TCGA-XX-A899  0.000000     0.0  \n",
       "TCGA-Z7-A8R5  0.000000     0.0  \n",
       "TCGA-Z7-A8R6  0.000000     0.0  \n",
       "\n",
       "[596 rows x 324 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_gene_matrix_xon17 = pd.read_csv(\"../data/processed/tcga_anno_features_xon17.csv\", index_col = 0)\n",
    "patient_gene_matrix_xon17"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30b99355-8b99-46d3-a218-7abeb7ef33dc",
   "metadata": {},
   "source": [
    "### Annovar annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "af5ca113",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation TP53 E286_E287del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRD4 X1340_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Y551Lfs*72 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR X20_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCORL1 G1178Efs*44 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CHEK1 T226Hfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FAM46C R286R - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 I143Dfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A R366Q - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMAD4 X263_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STAT3 X456_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 K565T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 A1079T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3R1 I381del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R342Efs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Y551Tfs*68 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM5A G1200Dfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH6 A34Pfs*47 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH6 F916* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN X342_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Q758Rfs*75 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCORL1 P1681Qfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CD79A R131Gfs*61 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FUBP1 S11Lfs*43 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 G1401Lfs*22 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 V654Cfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V97Sfs*26 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V157Sfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Q505Sfs*114 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CSF3R *837Yext*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X307_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Y948Hfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN1B H38Tfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A F2141Sfs*59 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRAF P403Lfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation INPP4B L93Ffs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 F313Lfs*67 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM X1833_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PBRM1 N258Kfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ALK H694Mfs*17 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA1 K223Rfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDK4 E206Dfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTCF M153* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB3 H1304Mfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation IKBKE R348Dfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation IRF2 N167Mfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MTAP A191Rfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3C2B R287Afs*92 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3R1 I82Sfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation POLE V2025* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PRKCI T276Nfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN S170Vfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN T319* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SOX9 K62del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SPEN K2609Rfs*29 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 K957T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN I33del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation AR X629_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM S214Ffs*40 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB4 X1061_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FH X413_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KEL X75_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 S1292S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRIP1 X381_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTCF S450Kfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A D394Y - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NTRK2 X573_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 N1198H - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 R551I - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CARD11 X647_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L194Efs*51 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM A78Efs*38 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CARD11 X1048_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 S313Tfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARAF S265L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR X2026_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation HGF X482_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 X550_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A X944_splice - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation MRE11A E466D - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A R388Q - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A D131N - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 R455R - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 E79* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PDGFRA X854_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3R1 X373_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RPTOR X169_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC1 X344_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 I503T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 P223* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GNAS S110G - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 Y236del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A G415G - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 A83Gfs*66 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MED12 X2164_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A M1318Cfs*163 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDC73 N401_E402del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRD4 E549Rfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NOTCH3 *2322Sext*106 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 E155Q - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 S239Cfs*61 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC X244_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 M642I - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN G20Dfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CARD11 E756del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FLT1 Q429Rfs*24 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 R692T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF2 X375_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 Q269E - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ALK Y479Tfs*28 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NOTCH3 E721Vfs*140 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 A827T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FBXW7 S668Vfs*39 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ESR1 G77Lfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A P318A - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 X703_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 R306Q - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CD22 X329_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 K860Nfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STK11 L282Afs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CASP8 X221_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CASP8 Q107Hfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GRM3 A317Pfs*78 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TGFBR2 X466_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EP300 N1236Kfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 X406_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 F650Lfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S299C - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S943Qfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC R1435Lfs*31 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 A629Pfs*71 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V173Efs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC D170Vfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SOX9 V291Afs*92 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC P1330Qfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF1 X2273_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SOX9 P237Tfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TGFBR2 H523Rfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC E1397Gfs*18 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 Q38Kfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC I1311Rfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM T1609* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM G2425Vfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S1163Tfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC I1574Yfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 S580S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN A126Lfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S1465Rfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCORL1 P1681Qfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EPHB1 P845Hfs*34 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB3 A805Hfs*95 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB4 M1? - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FGF23 G230Rfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NBN R551Gfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NBN E259del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 P37P - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TEK S586Vfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 P1343Qfs*49 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC L1509Efs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R280Kfs*59 - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation TP53 P60Qfs*63 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC T899Ifs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC T1556Nfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ACVR1B X421_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCOR L487Hfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC M1383Ifs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R290Kfs*53 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC H1490Nfs*25 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X224_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC D849Ifs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMAD4 A36_K45del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC V1414* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC L1488Yfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NPM1 X173_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation AKT3 X452_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 A693T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 C799C - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 N1090N - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 X706_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC A239Qfs*54 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation AR X629_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC L479Ifs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC T1313Nfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC N1531Rfs*31 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation HDAC1 E468del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S1465Wfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ASXL1 S846Qfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CHEK1 I234Nfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP I1084Nfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTNNB1 S45del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A N906Mfs*31 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation POLD1 X669_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMARCA4 G1136Afs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 P1281H - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC X178_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC E1552Qfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ETV5 X437_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TBX3 A147Sfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC H913Ifs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCOR R243H - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation POLE K778del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN X212_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC T1556Nfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S1465Wfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SOX9 G390Rfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation AR G461_G473del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FLT3 X205_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 Q144_L145dup - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 E286_E287del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S1205Tfs*60 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SOX9 G233Afs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCR X658_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FBXW7 L660Qfs*47 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMAD4 L540Pfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 X571_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A R1054Kfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SPOP X118_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STAG2 X699_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TBX3 M184Gfs*25 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L93Vfs*55 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X261_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDR Q1169Sfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 Y1258Ifs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 L731Ffs*39 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TBX3 E327Dfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA1 C903Wfs*97 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L299Afs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRIP1 X700_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN T319* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 D430G - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATRX D139Mfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X187_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA2 V1270del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF1 V2627Nfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GATA3 S402Yfs*107 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R156Afs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN1B T135Lfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 I251Sfs*94 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 V211Cfs*9 - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation WHSC1L1 S709T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 X522_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB2 L755S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 D764Gfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GATA3 R329Kfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GATA3 S410Qfs*97 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 S1077Qfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 N431Qfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 L380Kfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GATA3 R330Efs*23 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP2K4 Y113Lfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 P152Afs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA2 K437Ifs*22 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 C61Ifs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X33_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 T515_F516ins* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 S19Rfs*39 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 T522Ifs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 Y1276Mfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN D19Gfs*25 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATRX D678_E679del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A S617Qfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 S261* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GATA3 X308_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R156del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 N144Tfs*71 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 X722_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN W274Gfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 I304Nfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN V290Sfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3R1 K567_L570del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A V1113Gfs*38 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN I300Sfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EWSR1 S116_A117del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GATA3 H423Tfs*52 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MEN1 I85Yfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATRX X7_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 G352Rfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 T118Dfs*31 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 E553G - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 H1454Tfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 T1145Nfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 X1328_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A X960_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TBX3 K164Sfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 L71Sfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TBX3 X367_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TBX3 D160del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NOTCH1 X2060_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ASXL1 K539Sfs*164 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP Q355Tfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X367_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A R99Gfs*47 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A X153_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 D48Gfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A X153_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TGFBR2 D141del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A X51_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 H297Pfs*48 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EPHB4 A636Pfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDR X419_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L32Cfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FAM46C I263M - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X331_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 S37Pfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NOTCH1 G1710Afs*88 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 E444Q - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3R1 L573Qfs*27 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3CA N1068Kfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R337C - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRD4 K1250del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NOTCH1 D1124Afs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X367_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB2 V988I - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation P2RY8 V24Wfs*39 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X125_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FAS E141Nfs*46 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 P250Hfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 N200Ifs*47 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CARD11 X946_splice - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation TP53 C238Lfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NOTCH1 S647Pfs*123 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A S56Afs*90 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF2 X525_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TGFBR2 X466_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CASP8 X137_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 L326Rfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V272M - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X224_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R290Sfs*56 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FAM46C S158F - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V73Rfs*76 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R202_L206del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 S10I - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 X127_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA1 X1718_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP Q816Kfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RET R33Kfs*29 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A A36Rfs*17 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X32_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3R1 K575_R577del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X225_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 A189Pfs*58 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 V67F - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF1 E2155Vfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ALK R292H - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA2 L1390Ffs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN X267_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC1 A944Gfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A M54Ifs*66 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NBN L708* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CSF1R X585_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH6 A25_S28del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM Q1116Sfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NTRK3 X630_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V73Wfs*50 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A X51_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MDM2 H134Qfs*73 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation POLE X1576_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CHEK1 X272_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RAD51B X151_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V157Hfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR X2587_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NTRK3 E798Rfs*18 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 H178Pfs*69 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DOT1L N522Tfs*35 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB2 X633_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM5C E293Rfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NTRK2 K312Nfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PALB2 M875Cfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR X1315_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CYP17A1 X251_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC S1415Kfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TBX3 P224Qfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ALK X263_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GNAS P53Q - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 P419L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KEAP1 L175Wfs*55 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STK11 D53Gfs*110 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ROS1 X1557_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF1 P1951Hfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DOT1L X165_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STK11 X97_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FGF14 E151Kfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MET X861_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMARCA4 X1028_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X187_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SETD2 N571Kfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V73Wfs*50 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X331_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PDCD1LG2 X256_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A V115_A118del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB4 X572_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 R337C - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A M724Ifs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EGFR E746_A750del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A S700Lfs*29 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DOT1L A1069Pfs*31 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FGFR2 K420I - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB4 X208_splice - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation MRE11A Y111S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation P2RY8 R51Afs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RBM10 X387_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STK11 D53Tfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMARCA4 X1183_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STK11 Q220Pfs*46 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH6 K854del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X126_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PPP2R2A X153_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 E31V - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMARCA4 Y538Tfs*75 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FGF3 R171Afs*? - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 G355G - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STK11 A43Pfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM G2695del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EGFR E746_A750del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X331_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC2 X1044_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KEAP1 E540Rfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GABRA6 Q259Rfs*23 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STK11 E57Kfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A R1287Sfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Q1327Afs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCORL1 R743Afs*40 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA2 T3033Lfs*29 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 S70Pfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DDR1 R602Efs*51 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 K142Rfs*26 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P728Qfs*87 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CALR X234_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDK12 P277Sfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A I1625Mfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC N1797Kfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EPHA3 E647Rfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MLH1 K196Nfs*6 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P726Lfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SOX9 R257Gfs*23 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMAD4 Q388Cfs*28 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CIC P1116Qfs*45 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTCF T204Nfs*26 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EPHA3 M847Wfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PMS2 D414Tfs*34 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PRKCI E288Rfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SPEN I1052Sfs*40 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SPEN N2053Tfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TET2 L879del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR X98_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GNAS R75I - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 K174T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P726Sfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 I195Yfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A X601_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BARD1 K208Rfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCORL1 P1681Qfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CIC P509Hfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTNNA1 N738Ifs*53 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation IKBKE R348Dfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MEN1 E397Sfs*53 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH3 K383Rfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MYCN P45Rfs*86 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PALB2 N280Tfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 N428Mfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation QKI K134Rfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STAG2 M148Dfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC1 P362del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 I48Yfs*27 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 H178_S183del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A D1850Tfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR V543Cfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BARD1 K208Rfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 K860Nfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK2 X76_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 P343L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PPARG S186Vfs*47 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN N323Kfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 K589Ifs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 T419Pfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A L2281Ffs*104 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BARD1 D172Mfs*40 - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation BCORL1 P1681Qfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP N390Tfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation INPP4B R818Efs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 R1103Sfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 K860Nfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 K616del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 N339Ifs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH6 F1104Lfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PALB2 M296* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 V264I - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 E1099K - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 S910S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ACVR1B I305Lfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation AR Y531Lfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PRKCI Q205Pfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 E221Afs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN T167Nfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 C229Yfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation HNF1A L611Afs*38 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDH1 X337_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CASP8 R471_L474del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CBFB X95_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 I143Dfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P1898Hfs*25 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MEN1 R521Gfs*43 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMAD2 F402Lfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A X1705_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation QKI M1? - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF1 R262Lfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN T232Hfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A G276Efs*87 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A M890Hfs*46 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BARD1 K197Rfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DIS3 S445Hfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EZH2 F711Lfs*24 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARP2 X214_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A M890Hfs*46 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A D1850Tfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCOR L7Cfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCORL1 G893Afs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDK12 P683Qfs*70 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A A36Gfs*8 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP I1084Sfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTCF E691Rfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation POLD1 X751_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 P152Rfs*18 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTPRO N39del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM K2811Vfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA2 D946Rfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CASP8 V410Ffs*28 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CIC P509Hfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CIC L571Yfs*157 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTNNA1 F587Lfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MET K1199Sfs*49 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NTRK1 X452_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTCH1 R1308Efs*64 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC1 N891Tfs*40 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 L680L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 H826H - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Q758Rfs*75 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P1898Hfs*25 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARP1 S507Afs*17 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 M1335V - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ABL1 X304_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM S214Pfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATRX K2076Efs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCOR G400Afs*42 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCR D225Mfs*27 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTCF T204Kfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CYP17A1 Y60Ifs*29 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EED W73Mfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MYCN P45Rfs*86 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF1 N2470Mfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation POLE V551Ffs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN K267Rfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Q1327Afs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A F2141Sfs*59 - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation ATM T1399Pfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ETV4 R134Dfs*30 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EZR P119Lfs*34 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FANCL L191Wfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation IKBKE R49Afs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM5A D759Ifs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NBN F744Lfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PBRM1 I709Ffs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SGK1 F372Lfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC2 K34Nfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A D1850Tfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DIS3 K181Rfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERRFI1 N135Tfs*40 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FANCC I89Ffs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH3 K383Gfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation QKI K134Rfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SMARCB1 K364del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 P152Rfs*18 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 V73Wfs*50 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3C2G I437Nfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 P153Afs*28 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC T1556Nfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 R49Sfs*25 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ACVR1B G224Afs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Y551Lfs*72 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A D1850Tfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTCF T204Nfs*26 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DAXX C629Lfs*29 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DNMT3A K456Sfs*195 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB2 L755S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 K860Nfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN K164Rfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN K267Rfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SOX9 T478Hfs*100 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P1349Lfs*132 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA2 V1283Kfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRCA2 V1862* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EPHA3 X832_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FAS C135Vfs*52 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FGF19 L192Pfs*24 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FGFR2 N194Tfs*39 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation LYN E110Kfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MEN1 R521Gfs*43 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MLH1 R497Gfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH6 N897Ifs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SPEN A2251Qfs*102 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L35Cfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A D1850Tfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A F2141Sfs*59 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ASXL1 G645Vfs*58 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCOR K1207Nfs*31 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDK12 G1271Dfs*23 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP P2359del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERG F453Lfs*59 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FBXW7 S668Vfs*39 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM5A G8Afs*58 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A F1109Lfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PRKCI T276Qfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN W274Gfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN N323Mfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation XPO1 E908del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A L1591Cfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ASXL1 F1305Lfs*145 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation AXL H292Ifs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BARD1 D172Mfs*40 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCOR K839Sfs*17 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCORL1 P1681Qfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDK12 P683Qfs*70 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP T1593Pfs*42 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP I1084Sfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation DNMT3A P59Rfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ETV4 L48Sfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JAK1 L431Vfs*22 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MSH3 K383Rfs*32 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NOTCH1 E706Rfs*66 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 T240T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PAX5 V26Ffs*3 - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation PIK3C2B R287Afs*92 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation QKI K134Rfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ROS1 N1499Tfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TET2 V1006Gfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ZNF217 K525Nfs*23 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BCOR P1621Qfs*53 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NBN R466Gfs*18 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTCH1 N97Tfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 E278del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 P935S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P1115Qfs*46 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR K773Nfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CTNNB1 K335Nfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation JUN T95Hfs*12 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A N511Ifs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation POLE F699Vfs*11 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 G659Vfs*41 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RNF43 L311Sfs*108 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TEK P417Lfs*10 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TET2 L615Sfs*24 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 E294Sfs*51 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RPTOR X748_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM K2811Sfs*46 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FBXW7 T482Ifs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 T410T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 F45F - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 V144Lfs*9 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM X2325_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 A1130T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN R234Hfs*23 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FAM46C E261K - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAPK1 X242_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 K1311K - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRIP1 X831_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation SGK1 X377_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X125_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation BRIP1 K1040Dfs*4 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CASP8 E145Dfs*34 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MAP3K1 T946Nfs*57 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A X153_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FLT3 *994* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation MRE11A K196N - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 A379V - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3C2G E633Sfs*3 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 S84F - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation APC A703Qfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CBL I393_E394delinsK - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CCND1 E275_V281del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CIC S335F - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CYP17A1 R239Q - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation IKZF1 S445Rfs*15 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NF1 D2095Vfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 E353E - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 A1253A - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A S56Qfs*64 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB4 X956_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB4 S975L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GRM3 M1? - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation FGF6 M1? - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A S1182Rfs*24 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PRKCI T276Nfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Y500Lfs*123 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A Q117Rfs*63 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 T271Nfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STAG2 E1147Dfs*16 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ERBB2 L755S - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PARK2 V148L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P947Hfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 H825P - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN1A R84Gfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 X405_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ETV6 X155_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PBRM1 X989_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L350Pfs*17 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L348Wfs*22 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A H1124Yfs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDK12 S1377L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A N42Rfs*75 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A Q802Vfs*32 - No columns to parse from file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Encountered error while processing mutation KDM6A A756Qfs*18 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STAG2 X274_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PPARG T20Nfs*14 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN1A E108Gfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation EP300 E1334del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 L1184L - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation STAG2 E1187_D1192del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC1 X36_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN1A F51_D52ins* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN1A G40Afs*108 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PIK3C2B Q966Sfs*20 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 P195A - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CREBBP S1680del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A E2250Vfs*27 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTEN X343_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 A159Hfs*21 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 T155Sfs*13 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A M1318Dfs*5 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation PTPRO X369_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 X367_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A R1001* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 M708Kfs*2 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC1 X798_splice - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATM W1026Lfs*22 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN1A K16Gfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A P729Vfs*84 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CIC S313F - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 L508V - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1L1 R551T - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation CDKN2A A102Rfs*19 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A S2104Pfs*31 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 L191* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ARID1A S617* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation ATR N1135* - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 C141Afs*29 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation GATA3 M438Afs*38 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TP53 L35Ffs*7 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation TSC1 S183_A186del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation KDM6A E1050Kfs*33 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation RB1 R798Tfs*17 - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation NFE2L2 L30_S33del - No columns to parse from file\n",
      "ERROR:root:Encountered error while processing mutation WHSC1 K229E - No columns to parse from file\n"
     ]
    }
   ],
   "source": [
    "annotated_df = pd.DataFrame(columns=['input', 'sift_pred', 'sift4g_pred', 'lrt_pred', 'mutationtaster_pred',\n",
    "       'mutationassessor_pred', 'fathmm_pred', 'provean_pred', 'metasvm_pred',\n",
    "       'm_cap_pred', 'primateai_pred', 'deogen2_pred', 'bayesdel_addaf_pred',\n",
    "       'bayesdel_noaf_pred', 'clinpred_pred', 'list_s2_pred',\n",
    "       'fathmm_mkl_coding_pred', 'fathmm_xf_coding_pred', 'gene'])\n",
    "for submitter_id in tcga_mutations_filtered.submitter_id.unique():\n",
    "    filtered_df = tcga_mutations_filtered[\n",
    "        tcga_mutations_filtered.submitter_id == submitter_id\n",
    "    ]\n",
    "    point_mutations = list(filtered_df.point_mutation.values)\n",
    "    _, annot_df = construct_anno_features_xon17(submitter_id, point_mutations, False)\n",
    "    annotated_df = pd.concat([annotated_df, annot_df])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "0a57f951",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>primateai_pred</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FBXW7 R505G</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>FBXW7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HRAS E31K</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PPP2R1A S256F</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>PPP2R1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SPOP L282V</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SPOP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERBB3 F219V</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>ERBB3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           input sift_pred sift4g_pred lrt_pred mutationtaster_pred  \\\n",
       "0    FBXW7 R505G         1           1        1                   1   \n",
       "0      HRAS E31K         0           0        1                   1   \n",
       "1  PPP2R1A S256F         1           1        1                   1   \n",
       "2     SPOP L282V         1           1        1                   1   \n",
       "0    ERBB3 F219V         0           0        1                   1   \n",
       "\n",
       "  mutationassessor_pred fathmm_pred provean_pred metasvm_pred m_cap_pred  \\\n",
       "0                     1           0            1            0          1   \n",
       "0                     0           0            1            0          1   \n",
       "1                     1           0            1            0          1   \n",
       "2                     0           0            1            0          1   \n",
       "0                     1           0            1            1          1   \n",
       "\n",
       "  primateai_pred deogen2_pred bayesdel_addaf_pred bayesdel_noaf_pred  \\\n",
       "0              1            0                   1                  1   \n",
       "0              1            1                   1                  1   \n",
       "1              1            1                   1                  1   \n",
       "2              1            1                   1                  1   \n",
       "0              0            1                   1                  1   \n",
       "\n",
       "  clinpred_pred list_s2_pred fathmm_mkl_coding_pred fathmm_xf_coding_pred  \\\n",
       "0             1            1                      1                     1   \n",
       "0             1            0                      1                     1   \n",
       "1             1            1                      1                     1   \n",
       "2             1            1                      1                     1   \n",
       "0             1            1                      1                     1   \n",
       "\n",
       "      gene  \n",
       "0    FBXW7  \n",
       "0     HRAS  \n",
       "1  PPP2R1A  \n",
       "2     SPOP  \n",
       "0    ERBB3  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "310f618d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4532, 19)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "b353fff0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# annotated_df.to_csv(\"../data/processed/anno_features_per_mutation_tcga.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1efb35da",
   "metadata": {},
   "source": [
    "#### Clinvar annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "d6004925",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>primateai_pred</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FBXW7 R505G</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>FBXW7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HRAS E31K</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PPP2R1A S256F</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>PPP2R1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SPOP L282V</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SPOP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERBB3 F219V</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>ERBB3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KRAS G12D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KRAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NOTCH1 V2119M</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NOTCH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RB1 S834*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>RB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TP53 A161T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>TP53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>ZNF217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4532 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            input  sift_pred  sift4g_pred  lrt_pred  mutationtaster_pred  \\\n",
       "0     FBXW7 R505G          1            1         1                    1   \n",
       "0       HRAS E31K          0            0         1                    1   \n",
       "1   PPP2R1A S256F          1            1         1                    1   \n",
       "2      SPOP L282V          1            1         1                    1   \n",
       "0     ERBB3 F219V          0            0         1                    1   \n",
       "..            ...        ...          ...       ...                  ...   \n",
       "2       KRAS G12D          1            1         1                    1   \n",
       "3   NOTCH1 V2119M          1            0         0                    1   \n",
       "4       RB1 S834*          0            0         1                    1   \n",
       "5      TP53 A161T          1            1         1                    1   \n",
       "6    ZNF217 L231V          1            1         1                    1   \n",
       "\n",
       "    mutationassessor_pred  fathmm_pred  provean_pred  metasvm_pred  \\\n",
       "0                       1            0             1             0   \n",
       "0                       0            0             1             0   \n",
       "1                       1            0             1             0   \n",
       "2                       0            0             1             0   \n",
       "0                       1            0             1             1   \n",
       "..                    ...          ...           ...           ...   \n",
       "2                       1            0             1             1   \n",
       "3                       0            1             0             1   \n",
       "4                       0            0             0             0   \n",
       "5                       1            1             1             1   \n",
       "6                       1            0             1             0   \n",
       "\n",
       "    m_cap_pred  primateai_pred  deogen2_pred  bayesdel_addaf_pred  \\\n",
       "0            1               1             0                    1   \n",
       "0            1               1             1                    1   \n",
       "1            1               1             1                    1   \n",
       "2            1               1             1                    1   \n",
       "0            1               0             1                    1   \n",
       "..         ...             ...           ...                  ...   \n",
       "2            1               1             1                    1   \n",
       "3            1               0             0                    0   \n",
       "4            0               0             0                    1   \n",
       "5            1               0             1                    1   \n",
       "6            1               0             1                    1   \n",
       "\n",
       "    bayesdel_noaf_pred  clinpred_pred  list_s2_pred  fathmm_mkl_coding_pred  \\\n",
       "0                    1              1             1                       1   \n",
       "0                    1              1             0                       1   \n",
       "1                    1              1             1                       1   \n",
       "2                    1              1             1                       1   \n",
       "0                    1              1             1                       1   \n",
       "..                 ...            ...           ...                     ...   \n",
       "2                    1              1             0                       1   \n",
       "3                    0              1             1                       1   \n",
       "4                    1              0             0                       1   \n",
       "5                    1              1             1                       1   \n",
       "6                    0              1             1                       1   \n",
       "\n",
       "    fathmm_xf_coding_pred     gene  \n",
       "0                       1    FBXW7  \n",
       "0                       1     HRAS  \n",
       "1                       1  PPP2R1A  \n",
       "2                       1     SPOP  \n",
       "0                       1    ERBB3  \n",
       "..                    ...      ...  \n",
       "2                       1     KRAS  \n",
       "3                       1   NOTCH1  \n",
       "4                       0      RB1  \n",
       "5                       1     TP53  \n",
       "6                       0   ZNF217  \n",
       "\n",
       "[4532 rows x 19 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Also get Clinvar annotations for these mutations\n",
    "annotated_df = pd.read_csv(\"../data/processed/anno_features_per_mutation_tcga.csv\", index_col = 0)\n",
    "annotated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "0eff345a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ANNOTATION_SCRIPT_PATH_CLINVAR = \"../script/goAAtoGv2_clinvar.sh\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "d7c3875d",
   "metadata": {},
   "outputs": [],
   "source": [
    "results = []\n",
    "for mutation in list(annotated_df[\"input\"]):\n",
    "    try:\n",
    "        # Run annotation script within a temp file and extract features as DataFrame\n",
    "        with tempfile.TemporaryDirectory() as tmpdirname:\n",
    "            input_file_path = tmpdirname + \"anno_input.txt\"\n",
    "            with open(input_file_path, \"w+\") as input_file:\n",
    "                mutation_cleaned = [part for part in mutation.split(\" \") if part]\n",
    "                input_file.write(\":p.\".join(mutation_cleaned))\n",
    "                input_file.write(\"\\n\")\n",
    "\n",
    "            # Execute script\n",
    "            cmd = \"bash {0} {1}\".format(ANNOTATION_SCRIPT_PATH_CLINVAR, input_file_path)\n",
    "\n",
    "            logging.info(f\"Executing command {cmd}\")\n",
    "            subprocess.call(cmd, shell=True, executable=\"/bin/bash\")\n",
    "            out_file_path = f\"{input_file_path}.annot.hg38_finalannot.txt\"\n",
    "            res = pd.read_table(out_file_path)\n",
    "    # Some inputs lead to errors, such as \"PTEN loss\" - ignore and continue processing\n",
    "    except Exception as e:\n",
    "        logging.error(\n",
    "            f\"Encountered error while processing mutation {mutation} - {e}\"\n",
    "        )\n",
    "        continue\n",
    "    res[\"input\"] = mutation\n",
    "    res = res[~res.duplicated()]\n",
    "    results.append(res)\n",
    "clinvar_annot_df = pd.concat(results)\n",
    "clinvar_annot_df.set_index([\"input\"], inplace=True)\n",
    "clinvar_annot_df.drop(columns=[\"Otherinfo1\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "887ef837",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>SIFT_converted_rankscore</th>\n",
       "      <th>SIFT_pred</th>\n",
       "      <th>SIFT4G_score</th>\n",
       "      <th>SIFT4G_converted_rankscore</th>\n",
       "      <th>SIFT4G_pred</th>\n",
       "      <th>LRT_score</th>\n",
       "      <th>LRT_converted_rankscore</th>\n",
       "      <th>LRT_pred</th>\n",
       "      <th>MutationTaster_score</th>\n",
       "      <th>...</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>CLNALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>input</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>FBXW7 R505G</th>\n",
       "      <td>0.001</td>\n",
       "      <td>0.913</td>\n",
       "      <td>D</td>\n",
       "      <td>0.015</td>\n",
       "      <td>0.616</td>\n",
       "      <td>D</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.629</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.631</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>363302</td>\n",
       "      <td>B-cell_chronic_lymphocytic_leukemia|Transition...</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0005550,Human_Phen...</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HRAS E31K</th>\n",
       "      <td>0.109</td>\n",
       "      <td>0.311</td>\n",
       "      <td>T</td>\n",
       "      <td>0.345</td>\n",
       "      <td>0.177</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.843</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.338</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.826</td>\n",
       "      <td>0.344</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPP2R1A S256F</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.913</td>\n",
       "      <td>D</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.928</td>\n",
       "      <td>D</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.408</td>\n",
       "      <td>D</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.54</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.659</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPOP L282V</th>\n",
       "      <td>0.001</td>\n",
       "      <td>0.785</td>\n",
       "      <td>D</td>\n",
       "      <td>0.041</td>\n",
       "      <td>0.505</td>\n",
       "      <td>D</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.843</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.475</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERBB3 F219V</th>\n",
       "      <td>0.071</td>\n",
       "      <td>0.352</td>\n",
       "      <td>T</td>\n",
       "      <td>0.116</td>\n",
       "      <td>0.372</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.629</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.883</td>\n",
       "      <td>0.999</td>\n",
       "      <td>0.427</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KRAS G12D</th>\n",
       "      <td>0.01</td>\n",
       "      <td>0.565</td>\n",
       "      <td>D</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.928</td>\n",
       "      <td>D</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.843</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.88</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>27621</td>\n",
       "      <td>Vascular_Tumors_Including_Pyogenic_Granuloma|A...</td>\n",
       "      <td>.|Human_Phenotype_Ontology:HP:0001914,Human_Ph...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOTCH1 V2119M</th>\n",
       "      <td>0.047</td>\n",
       "      <td>0.403</td>\n",
       "      <td>D</td>\n",
       "      <td>0.062</td>\n",
       "      <td>0.453</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.629</td>\n",
       "      <td>N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.548</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.659</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RB1 S834*</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.629</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.894</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>420582</td>\n",
       "      <td>Retinoblastoma|Hereditary_cancer-predisposing_...</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0009919,MONDO:MOND...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53 A161T</th>\n",
       "      <td>0.006</td>\n",
       "      <td>0.913</td>\n",
       "      <td>D</td>\n",
       "      <td>0.013</td>\n",
       "      <td>0.631</td>\n",
       "      <td>D</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.843</td>\n",
       "      <td>D</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.569</td>\n",
       "      <td>0.999</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.524</td>\n",
       "      <td>171616</td>\n",
       "      <td>Li-Fraumeni_syndrome_1|Malignant_tumor_of_pros...</td>\n",
       "      <td>Gene:553989,MONDO:MONDO:0007903,MedGen:C183539...</td>\n",
       "      <td>criteria_provided,_conflicting_interpretations</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF217 L231V</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.913</td>\n",
       "      <td>D</td>\n",
       "      <td>0.009</td>\n",
       "      <td>0.668</td>\n",
       "      <td>D</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.629</td>\n",
       "      <td>D</td>\n",
       "      <td>0.999</td>\n",
       "      <td>...</td>\n",
       "      <td>0.887</td>\n",
       "      <td>0.995</td>\n",
       "      <td>0.388</td>\n",
       "      <td>0.938</td>\n",
       "      <td>0.406</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5109 rows × 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              SIFT_score SIFT_converted_rankscore SIFT_pred SIFT4G_score  \\\n",
       "input                                                                      \n",
       "FBXW7 R505G        0.001                    0.913         D        0.015   \n",
       "HRAS E31K          0.109                    0.311         T        0.345   \n",
       "PPP2R1A S256F        0.0                    0.913         D          0.0   \n",
       "SPOP L282V         0.001                    0.785         D        0.041   \n",
       "ERBB3 F219V        0.071                    0.352         T        0.116   \n",
       "...                  ...                      ...       ...          ...   \n",
       "KRAS G12D           0.01                    0.565         D          0.0   \n",
       "NOTCH1 V2119M      0.047                    0.403         D        0.062   \n",
       "RB1 S834*              .                        .         .            .   \n",
       "TP53 A161T         0.006                    0.913         D        0.013   \n",
       "ZNF217 L231V         0.0                    0.913         D        0.009   \n",
       "\n",
       "              SIFT4G_converted_rankscore SIFT4G_pred LRT_score  \\\n",
       "input                                                            \n",
       "FBXW7 R505G                        0.616           D       0.0   \n",
       "HRAS E31K                          0.177           T       0.0   \n",
       "PPP2R1A S256F                      0.928           D     0.001   \n",
       "SPOP L282V                         0.505           D       0.0   \n",
       "ERBB3 F219V                        0.372           T       0.0   \n",
       "...                                  ...         ...       ...   \n",
       "KRAS G12D                          0.928           D       0.0   \n",
       "NOTCH1 V2119M                      0.453           T       0.0   \n",
       "RB1 S834*                              .           .       0.0   \n",
       "TP53 A161T                         0.631           D       0.0   \n",
       "ZNF217 L231V                       0.668           D       0.0   \n",
       "\n",
       "              LRT_converted_rankscore LRT_pred MutationTaster_score  ...  \\\n",
       "input                                                                ...   \n",
       "FBXW7 R505G                     0.629        D                    1  ...   \n",
       "HRAS E31K                       0.843        D                    1  ...   \n",
       "PPP2R1A S256F                   0.408        D                  1.0  ...   \n",
       "SPOP L282V                      0.843        D                    1  ...   \n",
       "ERBB3 F219V                     0.629        D                    1  ...   \n",
       "...                               ...      ...                  ...  ...   \n",
       "KRAS G12D                       0.843        D                    1  ...   \n",
       "NOTCH1 V2119M                   0.629        N                  1.0  ...   \n",
       "RB1 S834*                       0.629        D                    1  ...   \n",
       "TP53 A161T                      0.843        D                  1.0  ...   \n",
       "ZNF217 L231V                    0.629        D                0.999  ...   \n",
       "\n",
       "              GERP++_RS_rankscore phastCons100way_vertebrate  \\\n",
       "input                                                          \n",
       "FBXW7 R505G                 0.631                        1.0   \n",
       "HRAS E31K                   0.338                        1.0   \n",
       "PPP2R1A S256F                0.54                        1.0   \n",
       "SPOP L282V                  0.475                        1.0   \n",
       "ERBB3 F219V                 0.883                      0.999   \n",
       "...                           ...                        ...   \n",
       "KRAS G12D                    0.88                        1.0   \n",
       "NOTCH1 V2119M               0.548                        1.0   \n",
       "RB1 S834*                   0.894                        1.0   \n",
       "TP53 A161T                  0.569                      0.999   \n",
       "ZNF217 L231V                0.887                      0.995   \n",
       "\n",
       "              phastCons100way_vertebrate_rankscore phastCons30way_mammalian  \\\n",
       "input                                                                         \n",
       "FBXW7 R505G                                  0.716                      1.0   \n",
       "HRAS E31K                                    0.716                    0.826   \n",
       "PPP2R1A S256F                                0.716                    0.998   \n",
       "SPOP L282V                                   0.716                      1.0   \n",
       "ERBB3 F219V                                  0.427                      1.0   \n",
       "...                                            ...                      ...   \n",
       "KRAS G12D                                    0.716                      1.0   \n",
       "NOTCH1 V2119M                                0.716                    0.998   \n",
       "RB1 S834*                                    0.716                      1.0   \n",
       "TP53 A161T                                   0.427                     0.99   \n",
       "ZNF217 L231V                                 0.388                    0.938   \n",
       "\n",
       "              phastCons30way_mammalian_rankscore CLNALLELEID  \\\n",
       "input                                                          \n",
       "FBXW7 R505G                                0.863      363302   \n",
       "HRAS E31K                                  0.344           .   \n",
       "PPP2R1A S256F                              0.659           .   \n",
       "SPOP L282V                                 0.863           .   \n",
       "ERBB3 F219V                                0.863           .   \n",
       "...                                          ...         ...   \n",
       "KRAS G12D                                  0.863       27621   \n",
       "NOTCH1 V2119M                              0.659           .   \n",
       "RB1 S834*                                  0.863      420582   \n",
       "TP53 A161T                                 0.524      171616   \n",
       "ZNF217 L231V                               0.406           .   \n",
       "\n",
       "                                                           CLNDN  \\\n",
       "input                                                              \n",
       "FBXW7 R505G    B-cell_chronic_lymphocytic_leukemia|Transition...   \n",
       "HRAS E31K                                                      .   \n",
       "PPP2R1A S256F                                                  .   \n",
       "SPOP L282V                                                     .   \n",
       "ERBB3 F219V                                                    .   \n",
       "...                                                          ...   \n",
       "KRAS G12D      Vascular_Tumors_Including_Pyogenic_Granuloma|A...   \n",
       "NOTCH1 V2119M                                                  .   \n",
       "RB1 S834*      Retinoblastoma|Hereditary_cancer-predisposing_...   \n",
       "TP53 A161T     Li-Fraumeni_syndrome_1|Malignant_tumor_of_pros...   \n",
       "ZNF217 L231V                                                   .   \n",
       "\n",
       "                                                        CLNDISDB  \\\n",
       "input                                                              \n",
       "FBXW7 R505G    Human_Phenotype_Ontology:HP:0005550,Human_Phen...   \n",
       "HRAS E31K                                                      .   \n",
       "PPP2R1A S256F                                                  .   \n",
       "SPOP L282V                                                     .   \n",
       "ERBB3 F219V                                                    .   \n",
       "...                                                          ...   \n",
       "KRAS G12D      .|Human_Phenotype_Ontology:HP:0001914,Human_Ph...   \n",
       "NOTCH1 V2119M                                                  .   \n",
       "RB1 S834*      Human_Phenotype_Ontology:HP:0009919,MONDO:MOND...   \n",
       "TP53 A161T     Gene:553989,MONDO:MONDO:0007903,MedGen:C183539...   \n",
       "ZNF217 L231V                                                   .   \n",
       "\n",
       "                                                      CLNREVSTAT  \\\n",
       "input                                                              \n",
       "FBXW7 R505G                       no_assertion_criteria_provided   \n",
       "HRAS E31K                                                      .   \n",
       "PPP2R1A S256F                                                  .   \n",
       "SPOP L282V                                                     .   \n",
       "ERBB3 F219V                                                    .   \n",
       "...                                                          ...   \n",
       "KRAS G12D      criteria_provided,_multiple_submitters,_no_con...   \n",
       "NOTCH1 V2119M                                                  .   \n",
       "RB1 S834*      criteria_provided,_multiple_submitters,_no_con...   \n",
       "TP53 A161T        criteria_provided,_conflicting_interpretations   \n",
       "ZNF217 L231V                                                   .   \n",
       "\n",
       "                                                     CLNSIG  \n",
       "input                                                        \n",
       "FBXW7 R505G                               Likely_pathogenic  \n",
       "HRAS E31K                                                 .  \n",
       "PPP2R1A S256F                                             .  \n",
       "SPOP L282V                                                .  \n",
       "ERBB3 F219V                                               .  \n",
       "...                                                     ...  \n",
       "KRAS G12D                                        Pathogenic  \n",
       "NOTCH1 V2119M                                             .  \n",
       "RB1 S834*                                        Pathogenic  \n",
       "TP53 A161T     Conflicting_interpretations_of_pathogenicity  \n",
       "ZNF217 L231V                                              .  \n",
       "\n",
       "[5109 rows x 73 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_annot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "5016e646",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".                                               3681\n",
       "Uncertain_significance                           475\n",
       "Pathogenic                                       461\n",
       "Pathogenic/Likely_pathogenic                     151\n",
       "Conflicting_interpretations_of_pathogenicity     135\n",
       "Likely_pathogenic                                130\n",
       "Pathogenic|drug_response|other                    22\n",
       "Likely_benign                                     21\n",
       "not_provided                                      11\n",
       "Benign                                             8\n",
       "Benign/Likely_benign                               8\n",
       "Pathogenic/Likely_pathogenic|other                 2\n",
       "drug_response                                      2\n",
       "Pathogenic/Likely_pathogenic|drug_response         1\n",
       "other                                              1\n",
       "Name: CLNSIG, dtype: int64"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_annot_df.CLNSIG.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "8bea2c7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_annot_df.to_csv(\"../data/processed/clinvar_anno_features_per_mutation_tcga.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d4d65ad4",
   "metadata": {},
   "source": [
    "### Variant Annotations from GPD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "30a1f60d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>point_mutation</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>primateai_pred</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "      <th>1plusxon17_score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-2E-A9G8</th>\n",
       "      <td>FBXW7 R505G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-2E-A9G8</th>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>AXL S447S</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AXL</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>BRD4 X1340_splice</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>HRAS E31K</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16831 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 point_mutation  sift_pred  sift4g_pred  lrt_pred  \\\n",
       "submitter_id                                                        \n",
       "TCGA-2E-A9G8        FBXW7 R505G        0.0          0.0       0.0   \n",
       "TCGA-2E-A9G8  TP53 E286_E287del        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH          AXL S447S        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH  BRD4 X1340_splice        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH          HRAS E31K        0.0          0.0       0.0   \n",
       "...                         ...        ...          ...       ...   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "\n",
       "              mutationtaster_pred  mutationassessor_pred  fathmm_pred  \\\n",
       "submitter_id                                                            \n",
       "TCGA-2E-A9G8                  0.0                    0.0          0.0   \n",
       "TCGA-2E-A9G8                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "...                           ...                    ...          ...   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "\n",
       "              provean_pred  metasvm_pred  m_cap_pred  primateai_pred  \\\n",
       "submitter_id                                                           \n",
       "TCGA-2E-A9G8           0.0           0.0         0.0             0.0   \n",
       "TCGA-2E-A9G8           0.0           0.0         0.0             0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0             0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0             0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0             0.0   \n",
       "...                    ...           ...         ...             ...   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0             0.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0             0.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0             0.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0             0.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0             0.0   \n",
       "\n",
       "              deogen2_pred  bayesdel_addaf_pred  bayesdel_noaf_pred  \\\n",
       "submitter_id                                                          \n",
       "TCGA-2E-A9G8           0.0                  0.0                 0.0   \n",
       "TCGA-2E-A9G8           0.0                  0.0                 0.0   \n",
       "TCGA-A5-A1OH           0.0                  0.0                 0.0   \n",
       "TCGA-A5-A1OH           0.0                  0.0                 0.0   \n",
       "TCGA-A5-A1OH           0.0                  0.0                 0.0   \n",
       "...                    ...                  ...                 ...   \n",
       "TCGA-XF-AAN7           1.0                  1.0                 0.0   \n",
       "TCGA-XF-AAN7           1.0                  1.0                 0.0   \n",
       "TCGA-XF-AAN7           1.0                  1.0                 0.0   \n",
       "TCGA-XF-AAN7           1.0                  1.0                 0.0   \n",
       "TCGA-XF-AAN7           1.0                  1.0                 0.0   \n",
       "\n",
       "              clinpred_pred  list_s2_pred  fathmm_mkl_coding_pred  \\\n",
       "submitter_id                                                        \n",
       "TCGA-2E-A9G8            0.0           0.0                     0.0   \n",
       "TCGA-2E-A9G8            0.0           0.0                     0.0   \n",
       "TCGA-A5-A1OH            0.0           0.0                     0.0   \n",
       "TCGA-A5-A1OH            0.0           0.0                     0.0   \n",
       "TCGA-A5-A1OH            0.0           0.0                     0.0   \n",
       "...                     ...           ...                     ...   \n",
       "TCGA-XF-AAN7            1.0           1.0                     1.0   \n",
       "TCGA-XF-AAN7            1.0           1.0                     1.0   \n",
       "TCGA-XF-AAN7            1.0           1.0                     1.0   \n",
       "TCGA-XF-AAN7            1.0           1.0                     1.0   \n",
       "TCGA-XF-AAN7            1.0           1.0                     1.0   \n",
       "\n",
       "              fathmm_xf_coding_pred    gene  1plusxon17_score  \n",
       "submitter_id                                                   \n",
       "TCGA-2E-A9G8                    0.0   FBXW7          1.000000  \n",
       "TCGA-2E-A9G8                    0.0    TP53          1.000000  \n",
       "TCGA-A5-A1OH                    0.0     AXL          1.000000  \n",
       "TCGA-A5-A1OH                    0.0    BRD4          1.000000  \n",
       "TCGA-A5-A1OH                    0.0    HRAS          1.000000  \n",
       "...                             ...     ...               ...  \n",
       "TCGA-XF-AAN7                    0.0  ZNF217          1.705882  \n",
       "TCGA-XF-AAN7                    0.0  ZNF217          1.705882  \n",
       "TCGA-XF-AAN7                    0.0  ZNF217          1.705882  \n",
       "TCGA-XF-AAN7                    0.0  ZNF217          1.705882  \n",
       "TCGA-XF-AAN7                    0.0  ZNF217          1.705882  \n",
       "\n",
       "[16831 rows x 20 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.read_csv(\"../data/processed/tcga_annovar_gpd_annot_per_patient_per_mutation.csv\", index_col = 0)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "18cffbed",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load intermediate GPD files for NPC and PC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "355f55d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>agg_sample_id</th>\n",
       "      <th>mut_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Hugo_Symbol, Gene, Chromosome, Start_Position, End_Position, Variant_Classification, Variant_Type, HGVSc, HGVSp, agg_sample_id, mut_freq]\n",
       "Index: []"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "npc_mutations = pd.read_csv(\"../data/processed/tcga_gpd_results/tcga_mutation_npc.tsv\", sep=\"\\t\")\n",
    "npc_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e00e2905",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: HGVSp, dtype: int64)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "npc_mutations[\"HGVSp\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0fbd6899",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: Variant_Classification, dtype: int64)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "npc_mutations[\"Variant_Classification\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3055a683",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Gene</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>prot_start_pos</th>\n",
       "      <th>prot_end_pos</th>\n",
       "      <th>agg_sample_id</th>\n",
       "      <th>mut_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7</td>\n",
       "      <td>140753336</td>\n",
       "      <td>140753336</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>ENSG00000157764</td>\n",
       "      <td>c.1799T&gt;A</td>\n",
       "      <td>p.V600E</td>\n",
       "      <td>600</td>\n",
       "      <td>600</td>\n",
       "      <td>TCGA-A6-2672_TCGA-A6-2686_TCGA-A6-3809_TCGA-A6...</td>\n",
       "      <td>259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>179218303</td>\n",
       "      <td>179218303</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>ENSG00000121879</td>\n",
       "      <td>c.1633G&gt;A</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>545</td>\n",
       "      <td>545</td>\n",
       "      <td>TCGA-A5-A1OJ_TCGA-AJ-A2QM_TCGA-AP-A0LN_TCGA-AP...</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>179234297</td>\n",
       "      <td>179234297</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>ENSG00000121879</td>\n",
       "      <td>c.3140A&gt;G</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>1047</td>\n",
       "      <td>1047</td>\n",
       "      <td>TCGA-AJ-A3BK_TCGA-AJ-A3NG_TCGA-AJ-A8CT_TCGA-AJ...</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>179218294</td>\n",
       "      <td>179218294</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>ENSG00000121879</td>\n",
       "      <td>c.1624G&gt;A</td>\n",
       "      <td>p.E542K</td>\n",
       "      <td>542</td>\n",
       "      <td>542</td>\n",
       "      <td>TCGA-4E-A92E_TCGA-A5-A0GM_TCGA-A5-A0GX_TCGA-A5...</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12</td>\n",
       "      <td>25245350</td>\n",
       "      <td>25245350</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>ENSG00000133703</td>\n",
       "      <td>c.35G&gt;A</td>\n",
       "      <td>p.G12D</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>TCGA-A5-A0GP_TCGA-AP-A0LD_TCGA-AP-A0LS_TCGA-AP...</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32208</th>\n",
       "      <td>8</td>\n",
       "      <td>37698543</td>\n",
       "      <td>37698543</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ZNF703</td>\n",
       "      <td>ENSG00000183779</td>\n",
       "      <td>c.1642T&gt;A</td>\n",
       "      <td>p.Y548N</td>\n",
       "      <td>548</td>\n",
       "      <td>548</td>\n",
       "      <td>TCGA-A7-A2KD</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32209</th>\n",
       "      <td>8</td>\n",
       "      <td>37698553</td>\n",
       "      <td>37698553</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ZNF703</td>\n",
       "      <td>ENSG00000183779</td>\n",
       "      <td>c.1652G&gt;T</td>\n",
       "      <td>p.S551I</td>\n",
       "      <td>551</td>\n",
       "      <td>551</td>\n",
       "      <td>TCGA-VQ-A8P2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32210</th>\n",
       "      <td>8</td>\n",
       "      <td>37698560</td>\n",
       "      <td>37698560</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ZNF703</td>\n",
       "      <td>ENSG00000183779</td>\n",
       "      <td>c.1659A&gt;T</td>\n",
       "      <td>p.L553F</td>\n",
       "      <td>553</td>\n",
       "      <td>553</td>\n",
       "      <td>TCGA-E6-A1LX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32211</th>\n",
       "      <td>8</td>\n",
       "      <td>37698604</td>\n",
       "      <td>37698604</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ZNF703</td>\n",
       "      <td>ENSG00000183779</td>\n",
       "      <td>c.1703delG</td>\n",
       "      <td>p.G568Dfs*14</td>\n",
       "      <td>568</td>\n",
       "      <td>568</td>\n",
       "      <td>TCGA-21-1070</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32212</th>\n",
       "      <td>8</td>\n",
       "      <td>37698649</td>\n",
       "      <td>37698649</td>\n",
       "      <td>Missense</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ZNF703</td>\n",
       "      <td>ENSG00000183779</td>\n",
       "      <td>c.1748C&gt;T</td>\n",
       "      <td>p.S583L</td>\n",
       "      <td>583</td>\n",
       "      <td>583</td>\n",
       "      <td>TCGA-DK-A6B6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32213 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chromosome  Start_Position  End_Position Variant_Classification  \\\n",
       "0              7       140753336     140753336               Missense   \n",
       "1              3       179218303     179218303               Missense   \n",
       "2              3       179234297     179234297               Missense   \n",
       "3              3       179218294     179218294               Missense   \n",
       "4             12        25245350      25245350               Missense   \n",
       "...          ...             ...           ...                    ...   \n",
       "32208          8        37698543      37698543               Missense   \n",
       "32209          8        37698553      37698553               Missense   \n",
       "32210          8        37698560      37698560               Missense   \n",
       "32211          8        37698604      37698604             Frameshift   \n",
       "32212          8        37698649      37698649               Missense   \n",
       "\n",
       "      Variant_Type Hugo_Symbol             Gene       HGVSc         HGVSp  \\\n",
       "0              SNP        BRAF  ENSG00000157764   c.1799T>A       p.V600E   \n",
       "1              SNP      PIK3CA  ENSG00000121879   c.1633G>A       p.E545K   \n",
       "2              SNP      PIK3CA  ENSG00000121879   c.3140A>G      p.H1047R   \n",
       "3              SNP      PIK3CA  ENSG00000121879   c.1624G>A       p.E542K   \n",
       "4              SNP        KRAS  ENSG00000133703     c.35G>A        p.G12D   \n",
       "...            ...         ...              ...         ...           ...   \n",
       "32208          SNP      ZNF703  ENSG00000183779   c.1642T>A       p.Y548N   \n",
       "32209          SNP      ZNF703  ENSG00000183779   c.1652G>T       p.S551I   \n",
       "32210          SNP      ZNF703  ENSG00000183779   c.1659A>T       p.L553F   \n",
       "32211          SNP      ZNF703  ENSG00000183779  c.1703delG  p.G568Dfs*14   \n",
       "32212          SNP      ZNF703  ENSG00000183779   c.1748C>T       p.S583L   \n",
       "\n",
       "       prot_start_pos  prot_end_pos  \\\n",
       "0                 600           600   \n",
       "1                 545           545   \n",
       "2                1047          1047   \n",
       "3                 542           542   \n",
       "4                  12            12   \n",
       "...               ...           ...   \n",
       "32208             548           548   \n",
       "32209             551           551   \n",
       "32210             553           553   \n",
       "32211             568           568   \n",
       "32212             583           583   \n",
       "\n",
       "                                           agg_sample_id  mut_freq  \n",
       "0      TCGA-A6-2672_TCGA-A6-2686_TCGA-A6-3809_TCGA-A6...       259  \n",
       "1      TCGA-A5-A1OJ_TCGA-AJ-A2QM_TCGA-AP-A0LN_TCGA-AP...       223  \n",
       "2      TCGA-AJ-A3BK_TCGA-AJ-A3NG_TCGA-AJ-A8CT_TCGA-AJ...       200  \n",
       "3      TCGA-4E-A92E_TCGA-A5-A0GM_TCGA-A5-A0GX_TCGA-A5...       150  \n",
       "4      TCGA-A5-A0GP_TCGA-AP-A0LD_TCGA-AP-A0LS_TCGA-AP...       118  \n",
       "...                                                  ...       ...  \n",
       "32208                                       TCGA-A7-A2KD         1  \n",
       "32209                                       TCGA-VQ-A8P2         1  \n",
       "32210                                       TCGA-E6-A1LX         1  \n",
       "32211                                       TCGA-21-1070         1  \n",
       "32212                                       TCGA-DK-A6B6         1  \n",
       "\n",
       "[32213 rows x 13 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pc_mutations = pd.read_csv(\"../data/processed/tcga_gpd_results/tcga_mutation_pc_pos.tsv\", sep=\"\\t\")\n",
    "pc_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7b0ae08d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Missense            26557\n",
       "Nonsense             3231\n",
       "Frameshift           2047\n",
       "InFrameDeletion       321\n",
       "MultiAAMissense        44\n",
       "InFrameInsertion       13\n",
       "Name: Variant_Classification, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pc_mutations[\"Variant_Classification\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ca8fdb6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot_accession</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>center_position</th>\n",
       "      <th>unit_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>unit_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A024RBG1</td>\n",
       "      <td>17</td>\n",
       "      <td>144</td>\n",
       "      <td>80</td>\n",
       "      <td>NUDIX</td>\n",
       "      <td>NUDT4</td>\n",
       "      <td>ENSG00000173598</td>\n",
       "      <td>Domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A024RBG1</td>\n",
       "      <td>17</td>\n",
       "      <td>144</td>\n",
       "      <td>80</td>\n",
       "      <td>NUDIX</td>\n",
       "      <td>NUDT4B</td>\n",
       "      <td>ENSG00000173598</td>\n",
       "      <td>Domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A024RBG1</td>\n",
       "      <td>17</td>\n",
       "      <td>144</td>\n",
       "      <td>80</td>\n",
       "      <td>NUDIX</td>\n",
       "      <td>NUDT4</td>\n",
       "      <td>ENSG00000177144</td>\n",
       "      <td>Domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A024RBG1</td>\n",
       "      <td>17</td>\n",
       "      <td>144</td>\n",
       "      <td>80</td>\n",
       "      <td>NUDIX</td>\n",
       "      <td>NUDT4B</td>\n",
       "      <td>ENSG00000177144</td>\n",
       "      <td>Domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A0A075B6H9</td>\n",
       "      <td>25</td>\n",
       "      <td>119</td>\n",
       "      <td>72</td>\n",
       "      <td>V-set</td>\n",
       "      <td>IGLV4-69</td>\n",
       "      <td>ENSG00000211637</td>\n",
       "      <td>Domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394261</th>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>196</td>\n",
       "      <td>206</td>\n",
       "      <td>201</td>\n",
       "      <td>acety</td>\n",
       "      <td>COLEC10</td>\n",
       "      <td>ENSG00000184374</td>\n",
       "      <td>PTM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394262</th>\n",
       "      <td>Q9Y6Z7</td>\n",
       "      <td>165</td>\n",
       "      <td>272</td>\n",
       "      <td>218</td>\n",
       "      <td>Lectin_C</td>\n",
       "      <td>COLEC10</td>\n",
       "      <td>ENSG00000184374</td>\n",
       "      <td>Domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394263</th>\n",
       "      <td>S4R3P1</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>Humanin</td>\n",
       "      <td>MTRNR2L13</td>\n",
       "      <td>ENSG00000270394</td>\n",
       "      <td>Family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394264</th>\n",
       "      <td>S4R3Y5</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>Humanin</td>\n",
       "      <td>MTRNR2L11</td>\n",
       "      <td>ENSG00000270188</td>\n",
       "      <td>Family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394265</th>\n",
       "      <td>U3KPV4</td>\n",
       "      <td>47</td>\n",
       "      <td>340</td>\n",
       "      <td>193</td>\n",
       "      <td>Glyco_transf_6</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>ENSG00000184389</td>\n",
       "      <td>Family</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>394265 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       uniprot_accession  start_position  end_position  center_position  \\\n",
       "1             A0A024RBG1              17           144               80   \n",
       "2             A0A024RBG1              17           144               80   \n",
       "3             A0A024RBG1              17           144               80   \n",
       "4             A0A024RBG1              17           144               80   \n",
       "5             A0A075B6H9              25           119               72   \n",
       "...                  ...             ...           ...              ...   \n",
       "394261            Q9Y6Z7             196           206              201   \n",
       "394262            Q9Y6Z7             165           272              218   \n",
       "394263            S4R3P1               1            24               12   \n",
       "394264            S4R3Y5               1            24               12   \n",
       "394265            U3KPV4              47           340              193   \n",
       "\n",
       "             unit_name  gene_name          gene_id unit_label  \n",
       "1                NUDIX      NUDT4  ENSG00000173598     Domain  \n",
       "2                NUDIX     NUDT4B  ENSG00000173598     Domain  \n",
       "3                NUDIX      NUDT4  ENSG00000177144     Domain  \n",
       "4                NUDIX     NUDT4B  ENSG00000177144     Domain  \n",
       "5                V-set   IGLV4-69  ENSG00000211637     Domain  \n",
       "...                ...        ...              ...        ...  \n",
       "394261           acety    COLEC10  ENSG00000184374        PTM  \n",
       "394262        Lectin_C    COLEC10  ENSG00000184374     Domain  \n",
       "394263         Humanin  MTRNR2L13  ENSG00000270394     Family  \n",
       "394264         Humanin  MTRNR2L11  ENSG00000270188     Family  \n",
       "394265  Glyco_transf_6    A3GALT2  ENSG00000184389     Family  \n",
       "\n",
       "[394265 rows x 8 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# To get PIU vs LU, we used the locations in ptm_pfam_combine.csv which is used in the GPD implementation\n",
    "ptm_pfam_df = pd.read_csv(\"/data/ajayago/druid/datasets/ptm_pfam_combine.csv\", index_col = 0)\n",
    "ptm_pfam_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d67aa256",
   "metadata": {},
   "outputs": [],
   "source": [
    "GPD_unit = []\n",
    "for idx, row in pc_mutations.iterrows():\n",
    "    subset_ptm = ptm_pfam_df[ptm_pfam_df.gene_id == row[\"Gene\"]]\n",
    "    x = \"LU\"\n",
    "    for idx, r in subset_ptm.iterrows():\n",
    "        if (row[\"prot_start_pos\"] >= r[\"start_position\"]) & (row[\"prot_start_pos\"] <= r[\"end_position\"]) \\\n",
    "    | (row[\"prot_end_pos\"] >= r[\"start_position\"]) & (row[\"prot_end_pos\"] <= r[\"end_position\"]):\n",
    "            x = \"PIU\"\n",
    "            break\n",
    "    GPD_unit.append(x)\n",
    "\n",
    "pc_mutations[\"GPD_unit\"] = GPD_unit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "501d9ba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PIU    21007\n",
       "LU     11206\n",
       "Name: GPD_unit, dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pc_mutations[\"GPD_unit\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "cfa208e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(21007, 11206)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "piu_mutations = set(pc_mutations[pc_mutations[\"GPD_unit\"] == \"PIU\"][\"Hugo_Symbol\"] + \" \" + pc_mutations[pc_mutations[\"GPD_unit\"] == \"PIU\"][\"HGVSp\"].apply(lambda x: x.split(\"p.\")[1]))\n",
    "lu_mutations = set(pc_mutations[pc_mutations[\"GPD_unit\"] == \"LU\"][\"Hugo_Symbol\"] + \" \" + pc_mutations[pc_mutations[\"GPD_unit\"] == \"LU\"][\"HGVSp\"].apply(lambda x: x.split(\"p.\")[1]))\n",
    "len(piu_mutations), len(lu_mutations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3bc5c1f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Map each point mutation to PIU/LU or NCU\n",
    "GPD_unit_merged1 = []\n",
    "for idx, row in merged_1.iterrows():\n",
    "    if row[\"point_mutation\"] in (piu_mutations):\n",
    "        GPD_unit_merged1.append(\"PIU\")\n",
    "    elif row[\"point_mutation\"] in (lu_mutations):\n",
    "        GPD_unit_merged1.append(\"LU\")\n",
    "    else:\n",
    "        GPD_unit_merged1.append(\"NCU\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "efff8c74",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_1[\"GPD_unit\"] = GPD_unit_merged1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c0deda8c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>point_mutation</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>...</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "      <th>1plusxon17_score</th>\n",
       "      <th>GPD_unit</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-2E-A9G8</th>\n",
       "      <td>FBXW7 R505G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PIU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-2E-A9G8</th>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PIU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>AXL S447S</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AXL</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NCU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>BRD4 X1340_splice</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NCU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>HRAS E31K</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PIU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16831 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 point_mutation  sift_pred  sift4g_pred  lrt_pred  \\\n",
       "submitter_id                                                        \n",
       "TCGA-2E-A9G8        FBXW7 R505G        0.0          0.0       0.0   \n",
       "TCGA-2E-A9G8  TP53 E286_E287del        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH          AXL S447S        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH  BRD4 X1340_splice        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH          HRAS E31K        0.0          0.0       0.0   \n",
       "...                         ...        ...          ...       ...   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "\n",
       "              mutationtaster_pred  mutationassessor_pred  fathmm_pred  \\\n",
       "submitter_id                                                            \n",
       "TCGA-2E-A9G8                  0.0                    0.0          0.0   \n",
       "TCGA-2E-A9G8                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "...                           ...                    ...          ...   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "\n",
       "              provean_pred  metasvm_pred  m_cap_pred  ...  deogen2_pred  \\\n",
       "submitter_id                                          ...                 \n",
       "TCGA-2E-A9G8           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-2E-A9G8           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0  ...           0.0   \n",
       "...                    ...           ...         ...  ...           ...   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "\n",
       "              bayesdel_addaf_pred  bayesdel_noaf_pred  clinpred_pred  \\\n",
       "submitter_id                                                           \n",
       "TCGA-2E-A9G8                  0.0                 0.0            0.0   \n",
       "TCGA-2E-A9G8                  0.0                 0.0            0.0   \n",
       "TCGA-A5-A1OH                  0.0                 0.0            0.0   \n",
       "TCGA-A5-A1OH                  0.0                 0.0            0.0   \n",
       "TCGA-A5-A1OH                  0.0                 0.0            0.0   \n",
       "...                           ...                 ...            ...   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "\n",
       "              list_s2_pred  fathmm_mkl_coding_pred  fathmm_xf_coding_pred  \\\n",
       "submitter_id                                                                \n",
       "TCGA-2E-A9G8           0.0                     0.0                    0.0   \n",
       "TCGA-2E-A9G8           0.0                     0.0                    0.0   \n",
       "TCGA-A5-A1OH           0.0                     0.0                    0.0   \n",
       "TCGA-A5-A1OH           0.0                     0.0                    0.0   \n",
       "TCGA-A5-A1OH           0.0                     0.0                    0.0   \n",
       "...                    ...                     ...                    ...   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "\n",
       "                gene 1plusxon17_score  GPD_unit  \n",
       "submitter_id                                     \n",
       "TCGA-2E-A9G8   FBXW7         1.000000       PIU  \n",
       "TCGA-2E-A9G8    TP53         1.000000       PIU  \n",
       "TCGA-A5-A1OH     AXL         1.000000       NCU  \n",
       "TCGA-A5-A1OH    BRD4         1.000000       NCU  \n",
       "TCGA-A5-A1OH    HRAS         1.000000       PIU  \n",
       "...              ...              ...       ...  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "\n",
       "[16831 rows x 21 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3ed927b",
   "metadata": {},
   "source": [
    "### Output: GPD + Annovar annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "286e9eba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>gene</th>\n",
       "      <th>ABL1</th>\n",
       "      <th>ACVR1B</th>\n",
       "      <th>AKT1</th>\n",
       "      <th>AKT2</th>\n",
       "      <th>AKT3</th>\n",
       "      <th>ALK</th>\n",
       "      <th>ALOX12B</th>\n",
       "      <th>APC</th>\n",
       "      <th>AR</th>\n",
       "      <th>ARAF</th>\n",
       "      <th>...</th>\n",
       "      <th>PARK2</th>\n",
       "      <th>PTEN</th>\n",
       "      <th>SDC4</th>\n",
       "      <th>SDHC</th>\n",
       "      <th>SOCS1</th>\n",
       "      <th>SRC</th>\n",
       "      <th>TERC</th>\n",
       "      <th>U2AF1</th>\n",
       "      <th>WHSC1</th>\n",
       "      <th>WHSC1L1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4384</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4390</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4398</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4427</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4432</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XX-A899</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>551 rows × 324 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "gene          ABL1  ACVR1B  AKT1  AKT2  AKT3  ALK  ALOX12B  APC   AR  ARAF  \\\n",
       "submitter_id                                                                 \n",
       "TCGA-05-4384   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4390   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4398   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4427   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4432   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "...            ...     ...   ...   ...   ...  ...      ...  ...  ...   ...   \n",
       "TCGA-XF-AAN5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XF-AAN7   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XX-A899   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-Z7-A8R5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-Z7-A8R6   0.0     0.0   0.0   0.0   0.0  1.0      0.0  0.0  0.0   0.0   \n",
       "\n",
       "gene          ...  PARK2  PTEN  SDC4  SDHC  SOCS1  SRC  TERC  U2AF1  WHSC1  \\\n",
       "submitter_id  ...                                                            \n",
       "TCGA-05-4384  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-05-4390  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-05-4398  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-05-4427  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-05-4432  ...      0     0     0     0      0    0     0      0      0   \n",
       "...           ...    ...   ...   ...   ...    ...  ...   ...    ...    ...   \n",
       "TCGA-XF-AAN5  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-XF-AAN7  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-XX-A899  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-Z7-A8R5  ...      0     0     0     0      0    0     0      0      0   \n",
       "TCGA-Z7-A8R6  ...      0     0     0     0      0    0     0      0      0   \n",
       "\n",
       "gene          WHSC1L1  \n",
       "submitter_id           \n",
       "TCGA-05-4384        0  \n",
       "TCGA-05-4390        0  \n",
       "TCGA-05-4398        0  \n",
       "TCGA-05-4427        0  \n",
       "TCGA-05-4432        0  \n",
       "...               ...  \n",
       "TCGA-XF-AAN5        0  \n",
       "TCGA-XF-AAN7        0  \n",
       "TCGA-XX-A899        0  \n",
       "TCGA-Z7-A8R5        0  \n",
       "TCGA-Z7-A8R6        0  \n",
       "\n",
       "[551 rows x 324 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_gene_matrix_xon17_piu = merged_1[merged_1.GPD_unit == \"PIU\"].pivot_table(index=\"submitter_id\", columns=\"gene\", values=\"1plusxon17_score\", aggfunc=\"max\")\n",
    "patient_gene_matrix_xon17_piu.fillna(0, inplace=True)\n",
    "for g in genes_324:\n",
    "    if g not in patient_gene_matrix_xon17_piu.columns:\n",
    "        patient_gene_matrix_xon17_piu[g] = 0\n",
    "\n",
    "patient_gene_matrix_xon17_piu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "4a3c9d6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>gene</th>\n",
       "      <th>ABL1</th>\n",
       "      <th>ACVR1B</th>\n",
       "      <th>AKT2</th>\n",
       "      <th>ALK</th>\n",
       "      <th>ALOX12B</th>\n",
       "      <th>APC</th>\n",
       "      <th>AR</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ASXL1</th>\n",
       "      <th>ATM</th>\n",
       "      <th>...</th>\n",
       "      <th>SOCS1</th>\n",
       "      <th>SPOP</th>\n",
       "      <th>SUFU</th>\n",
       "      <th>TERC</th>\n",
       "      <th>TGFBR2</th>\n",
       "      <th>U2AF1</th>\n",
       "      <th>VEGFA</th>\n",
       "      <th>VHL</th>\n",
       "      <th>WHSC1</th>\n",
       "      <th>WHSC1L1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4384</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4390</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4398</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4427</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4432</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>396 rows × 324 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "gene          ABL1  ACVR1B  AKT2  ALK  ALOX12B  APC   AR  ARID1A  ASXL1  ATM  \\\n",
       "submitter_id                                                                   \n",
       "TCGA-05-4384   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "TCGA-05-4390   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  1.0   \n",
       "TCGA-05-4398   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "TCGA-05-4427   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "TCGA-05-4432   0.0     0.0   0.0  1.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "...            ...     ...   ...  ...      ...  ...  ...     ...    ...  ...   \n",
       "TCGA-XF-AAN3   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "TCGA-XF-AAN5   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "TCGA-XF-AAN7   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "TCGA-Z7-A8R5   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "TCGA-Z7-A8R6   0.0     0.0   0.0  0.0      0.0  0.0  0.0     0.0    0.0  0.0   \n",
       "\n",
       "gene          ...  SOCS1  SPOP  SUFU  TERC  TGFBR2  U2AF1  VEGFA  VHL  WHSC1  \\\n",
       "submitter_id  ...                                                              \n",
       "TCGA-05-4384  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-05-4390  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-05-4398  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-05-4427  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-05-4432  ...      0     0     0     0       0      0      0    0      0   \n",
       "...           ...    ...   ...   ...   ...     ...    ...    ...  ...    ...   \n",
       "TCGA-XF-AAN3  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-XF-AAN5  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-XF-AAN7  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-Z7-A8R5  ...      0     0     0     0       0      0      0    0      0   \n",
       "TCGA-Z7-A8R6  ...      0     0     0     0       0      0      0    0      0   \n",
       "\n",
       "gene          WHSC1L1  \n",
       "submitter_id           \n",
       "TCGA-05-4384        0  \n",
       "TCGA-05-4390        0  \n",
       "TCGA-05-4398        0  \n",
       "TCGA-05-4427        0  \n",
       "TCGA-05-4432        0  \n",
       "...               ...  \n",
       "TCGA-XF-AAN3        0  \n",
       "TCGA-XF-AAN5        0  \n",
       "TCGA-XF-AAN7        0  \n",
       "TCGA-Z7-A8R5        0  \n",
       "TCGA-Z7-A8R6        0  \n",
       "\n",
       "[396 rows x 324 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_gene_matrix_xon17_lu = merged_1[merged_1.GPD_unit == \"LU\"].pivot_table(index=\"submitter_id\", columns=\"gene\", values=\"1plusxon17_score\", aggfunc=\"max\")\n",
    "patient_gene_matrix_xon17_lu.fillna(0, inplace=True)\n",
    "for g in genes_324:\n",
    "    if g not in patient_gene_matrix_xon17_lu.columns:\n",
    "        patient_gene_matrix_xon17_lu[g] = 0\n",
    "patient_gene_matrix_xon17_lu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e78ca37c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>gene</th>\n",
       "      <th>ABL1</th>\n",
       "      <th>ACVR1B</th>\n",
       "      <th>AKT1</th>\n",
       "      <th>AKT2</th>\n",
       "      <th>AKT3</th>\n",
       "      <th>ALK</th>\n",
       "      <th>ALOX12B</th>\n",
       "      <th>APC</th>\n",
       "      <th>AR</th>\n",
       "      <th>ARAF</th>\n",
       "      <th>...</th>\n",
       "      <th>RAD54L</th>\n",
       "      <th>SDHB</th>\n",
       "      <th>SDHC</th>\n",
       "      <th>SOCS1</th>\n",
       "      <th>SOX2</th>\n",
       "      <th>SUFU</th>\n",
       "      <th>TERC</th>\n",
       "      <th>U2AF1</th>\n",
       "      <th>VEGFA</th>\n",
       "      <th>ZNF703</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4384</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4398</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4402</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4427</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-05-4432</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XX-A899</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-Z7-A8R5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>389 rows × 324 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "gene          ABL1  ACVR1B  AKT1  AKT2  AKT3  ALK  ALOX12B  APC   AR  ARAF  \\\n",
       "submitter_id                                                                 \n",
       "TCGA-05-4384   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4398   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4402   0.0     0.0   0.0   0.0   0.0  0.0      0.0  1.0  0.0   0.0   \n",
       "TCGA-05-4427   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-05-4432   0.0     0.0   0.0   0.0   0.0  1.0      0.0  0.0  0.0   0.0   \n",
       "...            ...     ...   ...   ...   ...  ...      ...  ...  ...   ...   \n",
       "TCGA-XF-AAN3   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XF-AAN5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XF-AAN7   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-XX-A899   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "TCGA-Z7-A8R5   0.0     0.0   0.0   0.0   0.0  0.0      0.0  0.0  0.0   0.0   \n",
       "\n",
       "gene          ...  RAD54L  SDHB  SDHC  SOCS1  SOX2  SUFU  TERC  U2AF1  VEGFA  \\\n",
       "submitter_id  ...                                                              \n",
       "TCGA-05-4384  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-05-4398  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-05-4402  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-05-4427  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-05-4432  ...       0     0     0      0     0     0     0      0      0   \n",
       "...           ...     ...   ...   ...    ...   ...   ...   ...    ...    ...   \n",
       "TCGA-XF-AAN3  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-XF-AAN5  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-XF-AAN7  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-XX-A899  ...       0     0     0      0     0     0     0      0      0   \n",
       "TCGA-Z7-A8R5  ...       0     0     0      0     0     0     0      0      0   \n",
       "\n",
       "gene          ZNF703  \n",
       "submitter_id          \n",
       "TCGA-05-4384       0  \n",
       "TCGA-05-4398       0  \n",
       "TCGA-05-4402       0  \n",
       "TCGA-05-4427       0  \n",
       "TCGA-05-4432       0  \n",
       "...              ...  \n",
       "TCGA-XF-AAN3       0  \n",
       "TCGA-XF-AAN5       0  \n",
       "TCGA-XF-AAN7       0  \n",
       "TCGA-XX-A899       0  \n",
       "TCGA-Z7-A8R5       0  \n",
       "\n",
       "[389 rows x 324 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_gene_matrix_xon17_ncu = merged_1[merged_1.GPD_unit == \"NCU\"].pivot_table(index=\"submitter_id\", columns=\"gene\", values=\"1plusxon17_score\", aggfunc=\"max\")\n",
    "patient_gene_matrix_xon17_ncu.fillna(0, inplace=True)\n",
    "for g in genes_324:\n",
    "    if g not in patient_gene_matrix_xon17_ncu.columns:\n",
    "        patient_gene_matrix_xon17_ncu[g] = 0\n",
    "patient_gene_matrix_xon17_ncu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "38d0e9b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((551, 324), (396, 324), (389, 324))"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_gene_matrix_xon17_piu.shape, patient_gene_matrix_xon17_lu.shape, patient_gene_matrix_xon17_ncu.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9150a65b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(patient_gene_matrix_xon17.index) - set(patient_gene_matrix_xon17_piu.index))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "d2443870",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:9: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
      "<ipython-input-32-c3cae5a3b067>:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n"
     ]
    }
   ],
   "source": [
    "# Add in missing patient IDs in each matrix\n",
    "patient_gene_matrix_xon17_piu.reset_index(inplace=True)\n",
    "patient_gene_matrix_xon17_lu.reset_index(inplace=True)\n",
    "patient_gene_matrix_xon17_ncu.reset_index(inplace=True)\n",
    "for t in set(patient_gene_matrix_xon17.index) - set(patient_gene_matrix_xon17_piu.submitter_id):\n",
    "    patient_gene_matrix_xon17_piu = (patient_gene_matrix_xon17_piu.append({\"submitter_id\": t}, ignore_index=True))\n",
    "patient_gene_matrix_xon17_piu.set_index(\"submitter_id\", inplace=True)    \n",
    "for t in set(patient_gene_matrix_xon17.index) - set(patient_gene_matrix_xon17_lu.submitter_id):\n",
    "    patient_gene_matrix_xon17_lu = (patient_gene_matrix_xon17_lu.append({\"submitter_id\": t}, ignore_index=True))\n",
    "patient_gene_matrix_xon17_lu.set_index(\"submitter_id\", inplace=True)\n",
    "for t in set(patient_gene_matrix_xon17.index) - set(patient_gene_matrix_xon17_ncu.submitter_id):\n",
    "    patient_gene_matrix_xon17_ncu = (patient_gene_matrix_xon17_ncu.append({\"submitter_id\": t}, ignore_index=True))\n",
    "patient_gene_matrix_xon17_ncu.set_index(\"submitter_id\", inplace=True)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ea12e517",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((596, 324), (596, 324), (596, 324))"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patient_gene_matrix_xon17_piu.shape, patient_gene_matrix_xon17_lu.shape, patient_gene_matrix_xon17_ncu.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "9b0d4790",
   "metadata": {},
   "outputs": [],
   "source": [
    "patient_gene_matrix_xon17_piu.fillna(0, inplace=True)\n",
    "patient_gene_matrix_xon17_lu.fillna(0, inplace=True)\n",
    "patient_gene_matrix_xon17_ncu.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "a93180b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "patient_gene_matrix_xon17_piu.loc[patient_gene_matrix_xon17.index][genes_324].to_csv(\"../data/processed/xon17_gpd_annotations/tcga_piu_annotated_df.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "ccff9834",
   "metadata": {},
   "outputs": [],
   "source": [
    "patient_gene_matrix_xon17_lu.loc[patient_gene_matrix_xon17.index][genes_324].to_csv(\"../data/processed/xon17_gpd_annotations/tcga_lu_annotated_df.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a0635cab",
   "metadata": {},
   "outputs": [],
   "source": [
    "patient_gene_matrix_xon17_ncu.loc[patient_gene_matrix_xon17.index][genes_324].to_csv(\"../data/processed/xon17_gpd_annotations/tcga_ncu_annotated_df.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "532771f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_1.to_csv(\"../data/processed/tcga_annovar_gpd_annot_per_patient_per_mutation.csv\",)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "8937cd95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>point_mutation</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>...</th>\n",
       "      <th>deogen2_pred</th>\n",
       "      <th>bayesdel_addaf_pred</th>\n",
       "      <th>bayesdel_noaf_pred</th>\n",
       "      <th>clinpred_pred</th>\n",
       "      <th>list_s2_pred</th>\n",
       "      <th>fathmm_mkl_coding_pred</th>\n",
       "      <th>fathmm_xf_coding_pred</th>\n",
       "      <th>gene</th>\n",
       "      <th>1plusxon17_score</th>\n",
       "      <th>GPD_unit</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>submitter_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-2E-A9G8</th>\n",
       "      <td>FBXW7 R505G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PIU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-2E-A9G8</th>\n",
       "      <td>TP53 E286_E287del</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PIU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>AXL S447S</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AXL</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NCU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>BRD4 X1340_splice</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NCU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-A5-A1OH</th>\n",
       "      <td>HRAS E31K</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>PIU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-XF-AAN7</th>\n",
       "      <td>ZNF217 L231V</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ZNF217</td>\n",
       "      <td>1.705882</td>\n",
       "      <td>LU</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16831 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 point_mutation  sift_pred  sift4g_pred  lrt_pred  \\\n",
       "submitter_id                                                        \n",
       "TCGA-2E-A9G8        FBXW7 R505G        0.0          0.0       0.0   \n",
       "TCGA-2E-A9G8  TP53 E286_E287del        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH          AXL S447S        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH  BRD4 X1340_splice        0.0          0.0       0.0   \n",
       "TCGA-A5-A1OH          HRAS E31K        0.0          0.0       0.0   \n",
       "...                         ...        ...          ...       ...   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "TCGA-XF-AAN7       ZNF217 L231V        1.0          1.0       1.0   \n",
       "\n",
       "              mutationtaster_pred  mutationassessor_pred  fathmm_pred  \\\n",
       "submitter_id                                                            \n",
       "TCGA-2E-A9G8                  0.0                    0.0          0.0   \n",
       "TCGA-2E-A9G8                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "TCGA-A5-A1OH                  0.0                    0.0          0.0   \n",
       "...                           ...                    ...          ...   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "TCGA-XF-AAN7                  1.0                    1.0          0.0   \n",
       "\n",
       "              provean_pred  metasvm_pred  m_cap_pred  ...  deogen2_pred  \\\n",
       "submitter_id                                          ...                 \n",
       "TCGA-2E-A9G8           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-2E-A9G8           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0  ...           0.0   \n",
       "TCGA-A5-A1OH           0.0           0.0         0.0  ...           0.0   \n",
       "...                    ...           ...         ...  ...           ...   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "TCGA-XF-AAN7           1.0           0.0         1.0  ...           1.0   \n",
       "\n",
       "              bayesdel_addaf_pred  bayesdel_noaf_pred  clinpred_pred  \\\n",
       "submitter_id                                                           \n",
       "TCGA-2E-A9G8                  0.0                 0.0            0.0   \n",
       "TCGA-2E-A9G8                  0.0                 0.0            0.0   \n",
       "TCGA-A5-A1OH                  0.0                 0.0            0.0   \n",
       "TCGA-A5-A1OH                  0.0                 0.0            0.0   \n",
       "TCGA-A5-A1OH                  0.0                 0.0            0.0   \n",
       "...                           ...                 ...            ...   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "TCGA-XF-AAN7                  1.0                 0.0            1.0   \n",
       "\n",
       "              list_s2_pred  fathmm_mkl_coding_pred  fathmm_xf_coding_pred  \\\n",
       "submitter_id                                                                \n",
       "TCGA-2E-A9G8           0.0                     0.0                    0.0   \n",
       "TCGA-2E-A9G8           0.0                     0.0                    0.0   \n",
       "TCGA-A5-A1OH           0.0                     0.0                    0.0   \n",
       "TCGA-A5-A1OH           0.0                     0.0                    0.0   \n",
       "TCGA-A5-A1OH           0.0                     0.0                    0.0   \n",
       "...                    ...                     ...                    ...   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "TCGA-XF-AAN7           1.0                     1.0                    0.0   \n",
       "\n",
       "                gene 1plusxon17_score  GPD_unit  \n",
       "submitter_id                                     \n",
       "TCGA-2E-A9G8   FBXW7         1.000000       PIU  \n",
       "TCGA-2E-A9G8    TP53         1.000000       PIU  \n",
       "TCGA-A5-A1OH     AXL         1.000000       NCU  \n",
       "TCGA-A5-A1OH    BRD4         1.000000       NCU  \n",
       "TCGA-A5-A1OH    HRAS         1.000000       PIU  \n",
       "...              ...              ...       ...  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "TCGA-XF-AAN7  ZNF217         1.705882        LU  \n",
       "\n",
       "[16831 rows x 21 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d4ea7a90",
   "metadata": {},
   "source": [
    "#### Combine Clinvar annotations, GPD and Annovar annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "a4e530c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>point_mutation</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>...</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>CLNALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>input</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABL1 E189G</th>\n",
       "      <td>ABL1 E189G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.978</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.996</td>\n",
       "      <td>0.595</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABL1 E197*</th>\n",
       "      <td>ABL1 E197*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.978</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.991</td>\n",
       "      <td>0.532</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABL1 H246Y</th>\n",
       "      <td>ABL1 H246Y</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.978</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABL1 Q447P</th>\n",
       "      <td>ABL1 Q447P</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.706</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.995</td>\n",
       "      <td>0.577</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABL1 R533S</th>\n",
       "      <td>ABL1 R533S</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.819</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.948</td>\n",
       "      <td>0.416</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF217 V45A</th>\n",
       "      <td>ZNF217 V45A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.231</td>\n",
       "      <td>0.013</td>\n",
       "      <td>0.188</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.271</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF703 L63F</th>\n",
       "      <td>ZNF703 L63F</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.433</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.995</td>\n",
       "      <td>0.577</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF703 Q44*</th>\n",
       "      <td>ZNF703 Q44*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.441</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF703 S583L</th>\n",
       "      <td>ZNF703 S583L</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.395</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.978</td>\n",
       "      <td>0.47</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF703 Y548N</th>\n",
       "      <td>ZNF703 Y548N</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.416</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.716</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.863</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3855 rows × 94 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             point_mutation  sift_pred  sift4g_pred  lrt_pred  \\\n",
       "input                                                           \n",
       "ABL1 E189G       ABL1 E189G        0.0          0.0       0.0   \n",
       "ABL1 E197*       ABL1 E197*        0.0          0.0       0.0   \n",
       "ABL1 H246Y       ABL1 H246Y        0.0          0.0       0.0   \n",
       "ABL1 Q447P       ABL1 Q447P        0.0          0.0       0.0   \n",
       "ABL1 R533S       ABL1 R533S        0.0          0.0       0.0   \n",
       "...                     ...        ...          ...       ...   \n",
       "ZNF217 V45A     ZNF217 V45A        0.0          0.0       0.0   \n",
       "ZNF703 L63F     ZNF703 L63F        0.0          0.0       0.0   \n",
       "ZNF703 Q44*     ZNF703 Q44*        0.0          0.0       0.0   \n",
       "ZNF703 S583L   ZNF703 S583L        0.0          0.0       0.0   \n",
       "ZNF703 Y548N   ZNF703 Y548N        0.0          0.0       0.0   \n",
       "\n",
       "              mutationtaster_pred  mutationassessor_pred  fathmm_pred  \\\n",
       "input                                                                   \n",
       "ABL1 E189G                    0.0                    0.0          0.0   \n",
       "ABL1 E197*                    0.0                    0.0          0.0   \n",
       "ABL1 H246Y                    0.0                    0.0          0.0   \n",
       "ABL1 Q447P                    0.0                    0.0          0.0   \n",
       "ABL1 R533S                    0.0                    0.0          0.0   \n",
       "...                           ...                    ...          ...   \n",
       "ZNF217 V45A                   0.0                    0.0          0.0   \n",
       "ZNF703 L63F                   0.0                    0.0          0.0   \n",
       "ZNF703 Q44*                   0.0                    0.0          0.0   \n",
       "ZNF703 S583L                  0.0                    0.0          0.0   \n",
       "ZNF703 Y548N                  0.0                    0.0          0.0   \n",
       "\n",
       "              provean_pred  metasvm_pred  m_cap_pred  ...  \\\n",
       "input                                                 ...   \n",
       "ABL1 E189G             0.0           0.0         0.0  ...   \n",
       "ABL1 E197*             0.0           0.0         0.0  ...   \n",
       "ABL1 H246Y             0.0           0.0         0.0  ...   \n",
       "ABL1 Q447P             0.0           0.0         0.0  ...   \n",
       "ABL1 R533S             0.0           0.0         0.0  ...   \n",
       "...                    ...           ...         ...  ...   \n",
       "ZNF217 V45A            0.0           0.0         0.0  ...   \n",
       "ZNF703 L63F            0.0           0.0         0.0  ...   \n",
       "ZNF703 Q44*            0.0           0.0         0.0  ...   \n",
       "ZNF703 S583L           0.0           0.0         0.0  ...   \n",
       "ZNF703 Y548N           0.0           0.0         0.0  ...   \n",
       "\n",
       "              GERP++_RS_rankscore  phastCons100way_vertebrate  \\\n",
       "input                                                           \n",
       "ABL1 E189G                  0.978                         1.0   \n",
       "ABL1 E197*                  0.978                         1.0   \n",
       "ABL1 H246Y                  0.978                         1.0   \n",
       "ABL1 Q447P                  0.706                         1.0   \n",
       "ABL1 R533S                  0.049                       0.819   \n",
       "...                           ...                         ...   \n",
       "ZNF217 V45A                 0.231                       0.013   \n",
       "ZNF703 L63F                 0.433                         1.0   \n",
       "ZNF703 Q44*                 0.441                         1.0   \n",
       "ZNF703 S583L                0.395                         1.0   \n",
       "ZNF703 Y548N                0.416                         1.0   \n",
       "\n",
       "              phastCons100way_vertebrate_rankscore  phastCons30way_mammalian  \\\n",
       "input                                                                          \n",
       "ABL1 E189G                                   0.716                     0.996   \n",
       "ABL1 E197*                                   0.716                     0.991   \n",
       "ABL1 H246Y                                   0.716                       1.0   \n",
       "ABL1 Q447P                                   0.716                     0.995   \n",
       "ABL1 R533S                                     0.3                     0.948   \n",
       "...                                            ...                       ...   \n",
       "ZNF217 V45A                                  0.188                      0.49   \n",
       "ZNF703 L63F                                  0.716                     0.995   \n",
       "ZNF703 Q44*                                  0.716                       1.0   \n",
       "ZNF703 S583L                                 0.716                     0.978   \n",
       "ZNF703 Y548N                                 0.716                       1.0   \n",
       "\n",
       "              phastCons30way_mammalian_rankscore  CLNALLELEID  CLNDN  \\\n",
       "input                                                                  \n",
       "ABL1 E189G                                 0.595            .      .   \n",
       "ABL1 E197*                                 0.532            .      .   \n",
       "ABL1 H246Y                                 0.863            .      .   \n",
       "ABL1 Q447P                                 0.577            .      .   \n",
       "ABL1 R533S                                 0.416            .      .   \n",
       "...                                          ...          ...    ...   \n",
       "ZNF217 V45A                                0.271            .      .   \n",
       "ZNF703 L63F                                0.577            .      .   \n",
       "ZNF703 Q44*                                0.863            .      .   \n",
       "ZNF703 S583L                                0.47            .      .   \n",
       "ZNF703 Y548N                               0.863            .      .   \n",
       "\n",
       "              CLNDISDB CLNREVSTAT  CLNSIG  \n",
       "input                                      \n",
       "ABL1 E189G           .          .       .  \n",
       "ABL1 E197*           .          .       .  \n",
       "ABL1 H246Y           .          .       .  \n",
       "ABL1 Q447P           .          .       .  \n",
       "ABL1 R533S           .          .       .  \n",
       "...                ...        ...     ...  \n",
       "ZNF217 V45A          .          .       .  \n",
       "ZNF703 L63F          .          .       .  \n",
       "ZNF703 Q44*          .          .       .  \n",
       "ZNF703 S583L         .          .       .  \n",
       "ZNF703 Y548N         .          .       .  \n",
       "\n",
       "[3855 rows x 94 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_clinvar_df = pd.merge(merged_1, clinvar_annot_df.reset_index(), left_on=\"point_mutation\", right_on=\"input\").groupby(\"input\").aggregate(max)\n",
    "merged_clinvar_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "dfa927dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>point_mutation</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>sift4g_pred</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>m_cap_pred</th>\n",
       "      <th>...</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>CLNALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GPD_unit</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">LU</th>\n",
       "      <th>.</th>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>...</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "      <td>1008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Benign</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Benign/Likely_benign</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Conflicting_interpretations_of_pathogenicity</th>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Likely_benign</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Likely_pathogenic</th>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic</th>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>...</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic/Likely_pathogenic</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uncertain_significance</th>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>...</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>not_provided</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">NCU</th>\n",
       "      <th>.</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Benign/Likely_benign</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Conflicting_interpretations_of_pathogenicity</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Likely_benign</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>not_provided</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"15\" valign=\"top\">PIU</th>\n",
       "      <th>.</th>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>...</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "      <td>1906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Benign</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Benign/Likely_benign</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Conflicting_interpretations_of_pathogenicity</th>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>...</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Likely_benign</th>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Likely_pathogenic</th>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>...</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic</th>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>...</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "      <td>176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic/Likely_pathogenic</th>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>...</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic/Likely_pathogenic|drug_response</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic/Likely_pathogenic|other</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pathogenic|drug_response|other</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uncertain_significance</th>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>...</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "      <td>268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>drug_response</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>not_provided</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>other</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31 rows × 92 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       point_mutation  \\\n",
       "GPD_unit CLNSIG                                                         \n",
       "LU       .                                                       1008   \n",
       "         Benign                                                     1   \n",
       "         Benign/Likely_benign                                       2   \n",
       "         Conflicting_interpretations_of_pathogenicity              26   \n",
       "         Likely_benign                                              3   \n",
       "         Likely_pathogenic                                         18   \n",
       "         Pathogenic                                                63   \n",
       "         Pathogenic/Likely_pathogenic                              12   \n",
       "         Uncertain_significance                                   131   \n",
       "         not_provided                                               3   \n",
       "NCU      .                                                         25   \n",
       "         Benign/Likely_benign                                       1   \n",
       "         Conflicting_interpretations_of_pathogenicity               1   \n",
       "         Likely_benign                                              4   \n",
       "         Pathogenic                                                 1   \n",
       "         not_provided                                               1   \n",
       "PIU      .                                                       1906   \n",
       "         Benign                                                     3   \n",
       "         Benign/Likely_benign                                       1   \n",
       "         Conflicting_interpretations_of_pathogenicity              68   \n",
       "         Likely_benign                                             10   \n",
       "         Likely_pathogenic                                         55   \n",
       "         Pathogenic                                               176   \n",
       "         Pathogenic/Likely_pathogenic                              57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                 1   \n",
       "         Pathogenic/Likely_pathogenic|other                         2   \n",
       "         Pathogenic|drug_response|other                             1   \n",
       "         Uncertain_significance                                   268   \n",
       "         drug_response                                              1   \n",
       "         not_provided                                               5   \n",
       "         other                                                      1   \n",
       "\n",
       "                                                       sift_pred  sift4g_pred  \\\n",
       "GPD_unit CLNSIG                                                                 \n",
       "LU       .                                                  1008         1008   \n",
       "         Benign                                                1            1   \n",
       "         Benign/Likely_benign                                  2            2   \n",
       "         Conflicting_interpretations_of_pathogenicity         26           26   \n",
       "         Likely_benign                                         3            3   \n",
       "         Likely_pathogenic                                    18           18   \n",
       "         Pathogenic                                           63           63   \n",
       "         Pathogenic/Likely_pathogenic                         12           12   \n",
       "         Uncertain_significance                              131          131   \n",
       "         not_provided                                          3            3   \n",
       "NCU      .                                                    25           25   \n",
       "         Benign/Likely_benign                                  1            1   \n",
       "         Conflicting_interpretations_of_pathogenicity          1            1   \n",
       "         Likely_benign                                         4            4   \n",
       "         Pathogenic                                            1            1   \n",
       "         not_provided                                          1            1   \n",
       "PIU      .                                                  1906         1906   \n",
       "         Benign                                                3            3   \n",
       "         Benign/Likely_benign                                  1            1   \n",
       "         Conflicting_interpretations_of_pathogenicity         68           68   \n",
       "         Likely_benign                                        10           10   \n",
       "         Likely_pathogenic                                    55           55   \n",
       "         Pathogenic                                          176          176   \n",
       "         Pathogenic/Likely_pathogenic                         57           57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response            1            1   \n",
       "         Pathogenic/Likely_pathogenic|other                    2            2   \n",
       "         Pathogenic|drug_response|other                        1            1   \n",
       "         Uncertain_significance                              268          268   \n",
       "         drug_response                                         1            1   \n",
       "         not_provided                                          5            5   \n",
       "         other                                                 1            1   \n",
       "\n",
       "                                                       lrt_pred  \\\n",
       "GPD_unit CLNSIG                                                   \n",
       "LU       .                                                 1008   \n",
       "         Benign                                               1   \n",
       "         Benign/Likely_benign                                 2   \n",
       "         Conflicting_interpretations_of_pathogenicity        26   \n",
       "         Likely_benign                                        3   \n",
       "         Likely_pathogenic                                   18   \n",
       "         Pathogenic                                          63   \n",
       "         Pathogenic/Likely_pathogenic                        12   \n",
       "         Uncertain_significance                             131   \n",
       "         not_provided                                         3   \n",
       "NCU      .                                                   25   \n",
       "         Benign/Likely_benign                                 1   \n",
       "         Conflicting_interpretations_of_pathogenicity         1   \n",
       "         Likely_benign                                        4   \n",
       "         Pathogenic                                           1   \n",
       "         not_provided                                         1   \n",
       "PIU      .                                                 1906   \n",
       "         Benign                                               3   \n",
       "         Benign/Likely_benign                                 1   \n",
       "         Conflicting_interpretations_of_pathogenicity        68   \n",
       "         Likely_benign                                       10   \n",
       "         Likely_pathogenic                                   55   \n",
       "         Pathogenic                                         176   \n",
       "         Pathogenic/Likely_pathogenic                        57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response           1   \n",
       "         Pathogenic/Likely_pathogenic|other                   2   \n",
       "         Pathogenic|drug_response|other                       1   \n",
       "         Uncertain_significance                             268   \n",
       "         drug_response                                        1   \n",
       "         not_provided                                         5   \n",
       "         other                                                1   \n",
       "\n",
       "                                                       mutationtaster_pred  \\\n",
       "GPD_unit CLNSIG                                                              \n",
       "LU       .                                                            1008   \n",
       "         Benign                                                          1   \n",
       "         Benign/Likely_benign                                            2   \n",
       "         Conflicting_interpretations_of_pathogenicity                   26   \n",
       "         Likely_benign                                                   3   \n",
       "         Likely_pathogenic                                              18   \n",
       "         Pathogenic                                                     63   \n",
       "         Pathogenic/Likely_pathogenic                                   12   \n",
       "         Uncertain_significance                                        131   \n",
       "         not_provided                                                    3   \n",
       "NCU      .                                                              25   \n",
       "         Benign/Likely_benign                                            1   \n",
       "         Conflicting_interpretations_of_pathogenicity                    1   \n",
       "         Likely_benign                                                   4   \n",
       "         Pathogenic                                                      1   \n",
       "         not_provided                                                    1   \n",
       "PIU      .                                                            1906   \n",
       "         Benign                                                          3   \n",
       "         Benign/Likely_benign                                            1   \n",
       "         Conflicting_interpretations_of_pathogenicity                   68   \n",
       "         Likely_benign                                                  10   \n",
       "         Likely_pathogenic                                              55   \n",
       "         Pathogenic                                                    176   \n",
       "         Pathogenic/Likely_pathogenic                                   57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                      1   \n",
       "         Pathogenic/Likely_pathogenic|other                              2   \n",
       "         Pathogenic|drug_response|other                                  1   \n",
       "         Uncertain_significance                                        268   \n",
       "         drug_response                                                   1   \n",
       "         not_provided                                                    5   \n",
       "         other                                                           1   \n",
       "\n",
       "                                                       mutationassessor_pred  \\\n",
       "GPD_unit CLNSIG                                                                \n",
       "LU       .                                                              1008   \n",
       "         Benign                                                            1   \n",
       "         Benign/Likely_benign                                              2   \n",
       "         Conflicting_interpretations_of_pathogenicity                     26   \n",
       "         Likely_benign                                                     3   \n",
       "         Likely_pathogenic                                                18   \n",
       "         Pathogenic                                                       63   \n",
       "         Pathogenic/Likely_pathogenic                                     12   \n",
       "         Uncertain_significance                                          131   \n",
       "         not_provided                                                      3   \n",
       "NCU      .                                                                25   \n",
       "         Benign/Likely_benign                                              1   \n",
       "         Conflicting_interpretations_of_pathogenicity                      1   \n",
       "         Likely_benign                                                     4   \n",
       "         Pathogenic                                                        1   \n",
       "         not_provided                                                      1   \n",
       "PIU      .                                                              1906   \n",
       "         Benign                                                            3   \n",
       "         Benign/Likely_benign                                              1   \n",
       "         Conflicting_interpretations_of_pathogenicity                     68   \n",
       "         Likely_benign                                                    10   \n",
       "         Likely_pathogenic                                                55   \n",
       "         Pathogenic                                                      176   \n",
       "         Pathogenic/Likely_pathogenic                                     57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                        1   \n",
       "         Pathogenic/Likely_pathogenic|other                                2   \n",
       "         Pathogenic|drug_response|other                                    1   \n",
       "         Uncertain_significance                                          268   \n",
       "         drug_response                                                     1   \n",
       "         not_provided                                                      5   \n",
       "         other                                                             1   \n",
       "\n",
       "                                                       fathmm_pred  \\\n",
       "GPD_unit CLNSIG                                                      \n",
       "LU       .                                                    1008   \n",
       "         Benign                                                  1   \n",
       "         Benign/Likely_benign                                    2   \n",
       "         Conflicting_interpretations_of_pathogenicity           26   \n",
       "         Likely_benign                                           3   \n",
       "         Likely_pathogenic                                      18   \n",
       "         Pathogenic                                             63   \n",
       "         Pathogenic/Likely_pathogenic                           12   \n",
       "         Uncertain_significance                                131   \n",
       "         not_provided                                            3   \n",
       "NCU      .                                                      25   \n",
       "         Benign/Likely_benign                                    1   \n",
       "         Conflicting_interpretations_of_pathogenicity            1   \n",
       "         Likely_benign                                           4   \n",
       "         Pathogenic                                              1   \n",
       "         not_provided                                            1   \n",
       "PIU      .                                                    1906   \n",
       "         Benign                                                  3   \n",
       "         Benign/Likely_benign                                    1   \n",
       "         Conflicting_interpretations_of_pathogenicity           68   \n",
       "         Likely_benign                                          10   \n",
       "         Likely_pathogenic                                      55   \n",
       "         Pathogenic                                            176   \n",
       "         Pathogenic/Likely_pathogenic                           57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response              1   \n",
       "         Pathogenic/Likely_pathogenic|other                      2   \n",
       "         Pathogenic|drug_response|other                          1   \n",
       "         Uncertain_significance                                268   \n",
       "         drug_response                                           1   \n",
       "         not_provided                                            5   \n",
       "         other                                                   1   \n",
       "\n",
       "                                                       provean_pred  \\\n",
       "GPD_unit CLNSIG                                                       \n",
       "LU       .                                                     1008   \n",
       "         Benign                                                   1   \n",
       "         Benign/Likely_benign                                     2   \n",
       "         Conflicting_interpretations_of_pathogenicity            26   \n",
       "         Likely_benign                                            3   \n",
       "         Likely_pathogenic                                       18   \n",
       "         Pathogenic                                              63   \n",
       "         Pathogenic/Likely_pathogenic                            12   \n",
       "         Uncertain_significance                                 131   \n",
       "         not_provided                                             3   \n",
       "NCU      .                                                       25   \n",
       "         Benign/Likely_benign                                     1   \n",
       "         Conflicting_interpretations_of_pathogenicity             1   \n",
       "         Likely_benign                                            4   \n",
       "         Pathogenic                                               1   \n",
       "         not_provided                                             1   \n",
       "PIU      .                                                     1906   \n",
       "         Benign                                                   3   \n",
       "         Benign/Likely_benign                                     1   \n",
       "         Conflicting_interpretations_of_pathogenicity            68   \n",
       "         Likely_benign                                           10   \n",
       "         Likely_pathogenic                                       55   \n",
       "         Pathogenic                                             176   \n",
       "         Pathogenic/Likely_pathogenic                            57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response               1   \n",
       "         Pathogenic/Likely_pathogenic|other                       2   \n",
       "         Pathogenic|drug_response|other                           1   \n",
       "         Uncertain_significance                                 268   \n",
       "         drug_response                                            1   \n",
       "         not_provided                                             5   \n",
       "         other                                                    1   \n",
       "\n",
       "                                                       metasvm_pred  \\\n",
       "GPD_unit CLNSIG                                                       \n",
       "LU       .                                                     1008   \n",
       "         Benign                                                   1   \n",
       "         Benign/Likely_benign                                     2   \n",
       "         Conflicting_interpretations_of_pathogenicity            26   \n",
       "         Likely_benign                                            3   \n",
       "         Likely_pathogenic                                       18   \n",
       "         Pathogenic                                              63   \n",
       "         Pathogenic/Likely_pathogenic                            12   \n",
       "         Uncertain_significance                                 131   \n",
       "         not_provided                                             3   \n",
       "NCU      .                                                       25   \n",
       "         Benign/Likely_benign                                     1   \n",
       "         Conflicting_interpretations_of_pathogenicity             1   \n",
       "         Likely_benign                                            4   \n",
       "         Pathogenic                                               1   \n",
       "         not_provided                                             1   \n",
       "PIU      .                                                     1906   \n",
       "         Benign                                                   3   \n",
       "         Benign/Likely_benign                                     1   \n",
       "         Conflicting_interpretations_of_pathogenicity            68   \n",
       "         Likely_benign                                           10   \n",
       "         Likely_pathogenic                                       55   \n",
       "         Pathogenic                                             176   \n",
       "         Pathogenic/Likely_pathogenic                            57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response               1   \n",
       "         Pathogenic/Likely_pathogenic|other                       2   \n",
       "         Pathogenic|drug_response|other                           1   \n",
       "         Uncertain_significance                                 268   \n",
       "         drug_response                                            1   \n",
       "         not_provided                                             5   \n",
       "         other                                                    1   \n",
       "\n",
       "                                                       m_cap_pred  ...  \\\n",
       "GPD_unit CLNSIG                                                    ...   \n",
       "LU       .                                                   1008  ...   \n",
       "         Benign                                                 1  ...   \n",
       "         Benign/Likely_benign                                   2  ...   \n",
       "         Conflicting_interpretations_of_pathogenicity          26  ...   \n",
       "         Likely_benign                                          3  ...   \n",
       "         Likely_pathogenic                                     18  ...   \n",
       "         Pathogenic                                            63  ...   \n",
       "         Pathogenic/Likely_pathogenic                          12  ...   \n",
       "         Uncertain_significance                               131  ...   \n",
       "         not_provided                                           3  ...   \n",
       "NCU      .                                                     25  ...   \n",
       "         Benign/Likely_benign                                   1  ...   \n",
       "         Conflicting_interpretations_of_pathogenicity           1  ...   \n",
       "         Likely_benign                                          4  ...   \n",
       "         Pathogenic                                             1  ...   \n",
       "         not_provided                                           1  ...   \n",
       "PIU      .                                                   1906  ...   \n",
       "         Benign                                                 3  ...   \n",
       "         Benign/Likely_benign                                   1  ...   \n",
       "         Conflicting_interpretations_of_pathogenicity          68  ...   \n",
       "         Likely_benign                                         10  ...   \n",
       "         Likely_pathogenic                                     55  ...   \n",
       "         Pathogenic                                           176  ...   \n",
       "         Pathogenic/Likely_pathogenic                          57  ...   \n",
       "         Pathogenic/Likely_pathogenic|drug_response             1  ...   \n",
       "         Pathogenic/Likely_pathogenic|other                     2  ...   \n",
       "         Pathogenic|drug_response|other                         1  ...   \n",
       "         Uncertain_significance                               268  ...   \n",
       "         drug_response                                          1  ...   \n",
       "         not_provided                                           5  ...   \n",
       "         other                                                  1  ...   \n",
       "\n",
       "                                                       GERP++_RS  \\\n",
       "GPD_unit CLNSIG                                                    \n",
       "LU       .                                                  1008   \n",
       "         Benign                                                1   \n",
       "         Benign/Likely_benign                                  2   \n",
       "         Conflicting_interpretations_of_pathogenicity         26   \n",
       "         Likely_benign                                         3   \n",
       "         Likely_pathogenic                                    18   \n",
       "         Pathogenic                                           63   \n",
       "         Pathogenic/Likely_pathogenic                         12   \n",
       "         Uncertain_significance                              131   \n",
       "         not_provided                                          3   \n",
       "NCU      .                                                    25   \n",
       "         Benign/Likely_benign                                  1   \n",
       "         Conflicting_interpretations_of_pathogenicity          1   \n",
       "         Likely_benign                                         4   \n",
       "         Pathogenic                                            1   \n",
       "         not_provided                                          1   \n",
       "PIU      .                                                  1906   \n",
       "         Benign                                                3   \n",
       "         Benign/Likely_benign                                  1   \n",
       "         Conflicting_interpretations_of_pathogenicity         68   \n",
       "         Likely_benign                                        10   \n",
       "         Likely_pathogenic                                    55   \n",
       "         Pathogenic                                          176   \n",
       "         Pathogenic/Likely_pathogenic                         57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response            1   \n",
       "         Pathogenic/Likely_pathogenic|other                    2   \n",
       "         Pathogenic|drug_response|other                        1   \n",
       "         Uncertain_significance                              268   \n",
       "         drug_response                                         1   \n",
       "         not_provided                                          5   \n",
       "         other                                                 1   \n",
       "\n",
       "                                                       GERP++_RS_rankscore  \\\n",
       "GPD_unit CLNSIG                                                              \n",
       "LU       .                                                            1008   \n",
       "         Benign                                                          1   \n",
       "         Benign/Likely_benign                                            2   \n",
       "         Conflicting_interpretations_of_pathogenicity                   26   \n",
       "         Likely_benign                                                   3   \n",
       "         Likely_pathogenic                                              18   \n",
       "         Pathogenic                                                     63   \n",
       "         Pathogenic/Likely_pathogenic                                   12   \n",
       "         Uncertain_significance                                        131   \n",
       "         not_provided                                                    3   \n",
       "NCU      .                                                              25   \n",
       "         Benign/Likely_benign                                            1   \n",
       "         Conflicting_interpretations_of_pathogenicity                    1   \n",
       "         Likely_benign                                                   4   \n",
       "         Pathogenic                                                      1   \n",
       "         not_provided                                                    1   \n",
       "PIU      .                                                            1906   \n",
       "         Benign                                                          3   \n",
       "         Benign/Likely_benign                                            1   \n",
       "         Conflicting_interpretations_of_pathogenicity                   68   \n",
       "         Likely_benign                                                  10   \n",
       "         Likely_pathogenic                                              55   \n",
       "         Pathogenic                                                    176   \n",
       "         Pathogenic/Likely_pathogenic                                   57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                      1   \n",
       "         Pathogenic/Likely_pathogenic|other                              2   \n",
       "         Pathogenic|drug_response|other                                  1   \n",
       "         Uncertain_significance                                        268   \n",
       "         drug_response                                                   1   \n",
       "         not_provided                                                    5   \n",
       "         other                                                           1   \n",
       "\n",
       "                                                       phastCons100way_vertebrate  \\\n",
       "GPD_unit CLNSIG                                                                     \n",
       "LU       .                                                                   1008   \n",
       "         Benign                                                                 1   \n",
       "         Benign/Likely_benign                                                   2   \n",
       "         Conflicting_interpretations_of_pathogenicity                          26   \n",
       "         Likely_benign                                                          3   \n",
       "         Likely_pathogenic                                                     18   \n",
       "         Pathogenic                                                            63   \n",
       "         Pathogenic/Likely_pathogenic                                          12   \n",
       "         Uncertain_significance                                               131   \n",
       "         not_provided                                                           3   \n",
       "NCU      .                                                                     25   \n",
       "         Benign/Likely_benign                                                   1   \n",
       "         Conflicting_interpretations_of_pathogenicity                           1   \n",
       "         Likely_benign                                                          4   \n",
       "         Pathogenic                                                             1   \n",
       "         not_provided                                                           1   \n",
       "PIU      .                                                                   1906   \n",
       "         Benign                                                                 3   \n",
       "         Benign/Likely_benign                                                   1   \n",
       "         Conflicting_interpretations_of_pathogenicity                          68   \n",
       "         Likely_benign                                                         10   \n",
       "         Likely_pathogenic                                                     55   \n",
       "         Pathogenic                                                           176   \n",
       "         Pathogenic/Likely_pathogenic                                          57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                             1   \n",
       "         Pathogenic/Likely_pathogenic|other                                     2   \n",
       "         Pathogenic|drug_response|other                                         1   \n",
       "         Uncertain_significance                                               268   \n",
       "         drug_response                                                          1   \n",
       "         not_provided                                                           5   \n",
       "         other                                                                  1   \n",
       "\n",
       "                                                       phastCons100way_vertebrate_rankscore  \\\n",
       "GPD_unit CLNSIG                                                                               \n",
       "LU       .                                                                             1008   \n",
       "         Benign                                                                           1   \n",
       "         Benign/Likely_benign                                                             2   \n",
       "         Conflicting_interpretations_of_pathogenicity                                    26   \n",
       "         Likely_benign                                                                    3   \n",
       "         Likely_pathogenic                                                               18   \n",
       "         Pathogenic                                                                      63   \n",
       "         Pathogenic/Likely_pathogenic                                                    12   \n",
       "         Uncertain_significance                                                         131   \n",
       "         not_provided                                                                     3   \n",
       "NCU      .                                                                               25   \n",
       "         Benign/Likely_benign                                                             1   \n",
       "         Conflicting_interpretations_of_pathogenicity                                     1   \n",
       "         Likely_benign                                                                    4   \n",
       "         Pathogenic                                                                       1   \n",
       "         not_provided                                                                     1   \n",
       "PIU      .                                                                             1906   \n",
       "         Benign                                                                           3   \n",
       "         Benign/Likely_benign                                                             1   \n",
       "         Conflicting_interpretations_of_pathogenicity                                    68   \n",
       "         Likely_benign                                                                   10   \n",
       "         Likely_pathogenic                                                               55   \n",
       "         Pathogenic                                                                     176   \n",
       "         Pathogenic/Likely_pathogenic                                                    57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                                       1   \n",
       "         Pathogenic/Likely_pathogenic|other                                               2   \n",
       "         Pathogenic|drug_response|other                                                   1   \n",
       "         Uncertain_significance                                                         268   \n",
       "         drug_response                                                                    1   \n",
       "         not_provided                                                                     5   \n",
       "         other                                                                            1   \n",
       "\n",
       "                                                       phastCons30way_mammalian  \\\n",
       "GPD_unit CLNSIG                                                                   \n",
       "LU       .                                                                 1008   \n",
       "         Benign                                                               1   \n",
       "         Benign/Likely_benign                                                 2   \n",
       "         Conflicting_interpretations_of_pathogenicity                        26   \n",
       "         Likely_benign                                                        3   \n",
       "         Likely_pathogenic                                                   18   \n",
       "         Pathogenic                                                          63   \n",
       "         Pathogenic/Likely_pathogenic                                        12   \n",
       "         Uncertain_significance                                             131   \n",
       "         not_provided                                                         3   \n",
       "NCU      .                                                                   25   \n",
       "         Benign/Likely_benign                                                 1   \n",
       "         Conflicting_interpretations_of_pathogenicity                         1   \n",
       "         Likely_benign                                                        4   \n",
       "         Pathogenic                                                           1   \n",
       "         not_provided                                                         1   \n",
       "PIU      .                                                                 1906   \n",
       "         Benign                                                               3   \n",
       "         Benign/Likely_benign                                                 1   \n",
       "         Conflicting_interpretations_of_pathogenicity                        68   \n",
       "         Likely_benign                                                       10   \n",
       "         Likely_pathogenic                                                   55   \n",
       "         Pathogenic                                                         176   \n",
       "         Pathogenic/Likely_pathogenic                                        57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                           1   \n",
       "         Pathogenic/Likely_pathogenic|other                                   2   \n",
       "         Pathogenic|drug_response|other                                       1   \n",
       "         Uncertain_significance                                             268   \n",
       "         drug_response                                                        1   \n",
       "         not_provided                                                         5   \n",
       "         other                                                                1   \n",
       "\n",
       "                                                       phastCons30way_mammalian_rankscore  \\\n",
       "GPD_unit CLNSIG                                                                             \n",
       "LU       .                                                                           1008   \n",
       "         Benign                                                                         1   \n",
       "         Benign/Likely_benign                                                           2   \n",
       "         Conflicting_interpretations_of_pathogenicity                                  26   \n",
       "         Likely_benign                                                                  3   \n",
       "         Likely_pathogenic                                                             18   \n",
       "         Pathogenic                                                                    63   \n",
       "         Pathogenic/Likely_pathogenic                                                  12   \n",
       "         Uncertain_significance                                                       131   \n",
       "         not_provided                                                                   3   \n",
       "NCU      .                                                                             25   \n",
       "         Benign/Likely_benign                                                           1   \n",
       "         Conflicting_interpretations_of_pathogenicity                                   1   \n",
       "         Likely_benign                                                                  4   \n",
       "         Pathogenic                                                                     1   \n",
       "         not_provided                                                                   1   \n",
       "PIU      .                                                                           1906   \n",
       "         Benign                                                                         3   \n",
       "         Benign/Likely_benign                                                           1   \n",
       "         Conflicting_interpretations_of_pathogenicity                                  68   \n",
       "         Likely_benign                                                                 10   \n",
       "         Likely_pathogenic                                                             55   \n",
       "         Pathogenic                                                                   176   \n",
       "         Pathogenic/Likely_pathogenic                                                  57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response                                     1   \n",
       "         Pathogenic/Likely_pathogenic|other                                             2   \n",
       "         Pathogenic|drug_response|other                                                 1   \n",
       "         Uncertain_significance                                                       268   \n",
       "         drug_response                                                                  1   \n",
       "         not_provided                                                                   5   \n",
       "         other                                                                          1   \n",
       "\n",
       "                                                       CLNALLELEID  CLNDN  \\\n",
       "GPD_unit CLNSIG                                                             \n",
       "LU       .                                                    1008   1008   \n",
       "         Benign                                                  1      1   \n",
       "         Benign/Likely_benign                                    2      2   \n",
       "         Conflicting_interpretations_of_pathogenicity           26     26   \n",
       "         Likely_benign                                           3      3   \n",
       "         Likely_pathogenic                                      18     18   \n",
       "         Pathogenic                                             63     63   \n",
       "         Pathogenic/Likely_pathogenic                           12     12   \n",
       "         Uncertain_significance                                131    131   \n",
       "         not_provided                                            3      3   \n",
       "NCU      .                                                      25     25   \n",
       "         Benign/Likely_benign                                    1      1   \n",
       "         Conflicting_interpretations_of_pathogenicity            1      1   \n",
       "         Likely_benign                                           4      4   \n",
       "         Pathogenic                                              1      1   \n",
       "         not_provided                                            1      1   \n",
       "PIU      .                                                    1906   1906   \n",
       "         Benign                                                  3      3   \n",
       "         Benign/Likely_benign                                    1      1   \n",
       "         Conflicting_interpretations_of_pathogenicity           68     68   \n",
       "         Likely_benign                                          10     10   \n",
       "         Likely_pathogenic                                      55     55   \n",
       "         Pathogenic                                            176    176   \n",
       "         Pathogenic/Likely_pathogenic                           57     57   \n",
       "         Pathogenic/Likely_pathogenic|drug_response              1      1   \n",
       "         Pathogenic/Likely_pathogenic|other                      2      2   \n",
       "         Pathogenic|drug_response|other                          1      1   \n",
       "         Uncertain_significance                                268    268   \n",
       "         drug_response                                           1      1   \n",
       "         not_provided                                            5      5   \n",
       "         other                                                   1      1   \n",
       "\n",
       "                                                       CLNDISDB  CLNREVSTAT  \n",
       "GPD_unit CLNSIG                                                              \n",
       "LU       .                                                 1008        1008  \n",
       "         Benign                                               1           1  \n",
       "         Benign/Likely_benign                                 2           2  \n",
       "         Conflicting_interpretations_of_pathogenicity        26          26  \n",
       "         Likely_benign                                        3           3  \n",
       "         Likely_pathogenic                                   18          18  \n",
       "         Pathogenic                                          63          63  \n",
       "         Pathogenic/Likely_pathogenic                        12          12  \n",
       "         Uncertain_significance                             131         131  \n",
       "         not_provided                                         3           3  \n",
       "NCU      .                                                   25          25  \n",
       "         Benign/Likely_benign                                 1           1  \n",
       "         Conflicting_interpretations_of_pathogenicity         1           1  \n",
       "         Likely_benign                                        4           4  \n",
       "         Pathogenic                                           1           1  \n",
       "         not_provided                                         1           1  \n",
       "PIU      .                                                 1906        1906  \n",
       "         Benign                                               3           3  \n",
       "         Benign/Likely_benign                                 1           1  \n",
       "         Conflicting_interpretations_of_pathogenicity        68          68  \n",
       "         Likely_benign                                       10          10  \n",
       "         Likely_pathogenic                                   55          55  \n",
       "         Pathogenic                                         176         176  \n",
       "         Pathogenic/Likely_pathogenic                        57          57  \n",
       "         Pathogenic/Likely_pathogenic|drug_response           1           1  \n",
       "         Pathogenic/Likely_pathogenic|other                   2           2  \n",
       "         Pathogenic|drug_response|other                       1           1  \n",
       "         Uncertain_significance                             268         268  \n",
       "         drug_response                                        1           1  \n",
       "         not_provided                                         5           5  \n",
       "         other                                                1           1  \n",
       "\n",
       "[31 rows x 92 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_clinvar_df.groupby([\"GPD_unit\", \"CLNSIG\"]).aggregate(\"count\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3408b69c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base2",
   "language": "python",
   "name": "base2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
